// @wix/cli/build/chunk-LGNVOZGE.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useBackendAsAServiceClient
} from "./chunk-SKN7SRA5.js";
import {
  ENV_FILE_NAME,
  getEnvFilePath,
  useProjectModel
} from "./chunk-OPNNOJNG.js";
import {
  SiteAuthProvider
} from "./chunk-3Q7XZST5.js";
import {
  Alert,
  Spinner,
  useAsync
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import {
  updateEnvFile
} from "./chunk-2OYCKIQP.js";
import {
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/EnvPullCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/EnvPullCommand/EnvPullCommand.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/EnvPullCommand/EnvPull.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var ENV_NAME = "prod";
var EnvPull = () => {
  const { model } = useProjectModel();
  const backendAsAServiceClient = useBackendAsAServiceClient();
  const { status } = useAsync(async () => {
    const appEnvironment = await backendAsAServiceClient.getAppEnvironmentByName(
      model.config.appId,
      ENV_NAME
    );
    await updateEnvFile(
      getEnvFilePath(model.projectFolder),
      appEnvironment.variables
    );
  }, []);
  return (0, import_variant.match)(status, {
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: `Pulling environment "${ENV_NAME}" variables...` }),
    Success: () => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, 'Environment variables written to "', ENV_FILE_NAME, '"'),
    Error: () => null,
    NotRequested: () => null
  });
};

// ../cli-astro-commands/src/components/EnvPullCommand/EnvPullCommand.tsx
var EnvPullCommand = () => {
  const {
    model: { config }
  } = useProjectModel();
  let commandTree = /* @__PURE__ */ import_react2.default.createElement(EnvPull, null);
  if (config.projectType === "Site") {
    commandTree = /* @__PURE__ */ import_react2.default.createElement(SiteAuthProvider, { siteId: config.siteId }, commandTree);
  }
  return commandTree;
};

export {
  EnvPullCommand
};
