// @wix/cli/build/chunk-6EJNAOBP.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  external_exports
} from "./chunk-F43XHKVL.js";
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site-old-definitions/src/index.ts
init_esm_shims();
import { join } from "node:path";
var VELO_DEPENDENCIES_CONFIG_FILENAME = "velo.dependencies.json";
var VELO_DEPENDENCIES_CONFIG_PATH = join(
  "src",
  VELO_DEPENDENCIES_CONFIG_FILENAME
);
var veloConfigSchema = external_exports.object({
  siteId: external_exports.string().uuid(),
  uiVersion: external_exports.string()
});
var veloDependenciesConfigSchema = external_exports.object({
  dependencies: external_exports.record(external_exports.string())
});
function getVeloDependenciesConfigPath(projectFolder) {
  return join(projectFolder, VELO_DEPENDENCIES_CONFIG_PATH);
}

export {
  VELO_DEPENDENCIES_CONFIG_FILENAME,
  VELO_DEPENDENCIES_CONFIG_PATH,
  veloConfigSchema,
  veloDependenciesConfigSchema,
  getVeloDependenciesConfigPath
};
