// @wix/cli/build/chunk-SC2W3WT3.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../../node_modules/@wix/bi-logger-dev-tools-data/dist/esm/v2/index.js
init_esm_shims();
function wixCliAppFlowFileUploadStarted(params) {
  return { evid: 5208, src: 11, endpoint: "", params };
}
function wixCliAppFlowUploadEnded(params) {
  return { evid: 5210, src: 11, endpoint: "", params };
}
function wixCliAppFlowStepsAnswers(params) {
  return { evid: 5229, src: 11, endpoint: "", params };
}
function wixCliAction(params) {
  return { evid: 5231, src: 11, endpoint: "", params };
}
function wixCliAppsFlowDeploymentStatusSrc11Evid5232(params) {
  return { evid: 5232, src: 11, endpoint: "", params };
}
function wixCliAppsFlowTunnelCreationStatusSrc11Evid5233(params) {
  return { evid: 5233, src: 11, endpoint: "", params };
}

export {
  wixCliAppFlowFileUploadStarted,
  wixCliAppFlowUploadEnded,
  wixCliAppFlowStepsAnswers,
  wixCliAction,
  wixCliAppsFlowDeploymentStatusSrc11Evid5232,
  wixCliAppsFlowTunnelCreationStatusSrc11Evid5233
};
