// @wix/cli/build/chunk-FUJVYDK4.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  ProjectModelProvider,
  loadProjectModel
} from "./chunk-OPNNOJNG.js";
import {
  HttpClientProvider,
  buildHttpClientHeaders
} from "./chunk-RHVTQ2DQ.js";
import {
  PanoramaProvider
} from "./chunk-RDT7NRVZ.js";
import {
  addUserFields,
  reportCommandStartEvent
} from "./chunk-ZWLXMEDA.js";
import {
  AccountAuthProvider,
  BiProvider,
  DebugLogProvider,
  ErrorReporterProvider
} from "./chunk-3Q7XZST5.js";
import {
  render
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import {
  getDebugLogFilePath
} from "./chunk-TKJGN2XT.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/commands/render-command.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
import { cwd } from "node:process";
function getFlow(model) {
  return model.config.projectType === "App" ? "Astro_App" : "Headless";
}
async function initServices(model, { biLogger, errorReporter }) {
  errorReporter.setContext("Application Info", {
    "Application ID": model.config.appId,
    "Project Type": model.config.projectType,
    ...model.config.projectType === "Site" ? {
      "Site ID": model.config.siteId
    } : {}
  });
  biLogger.updateDefaults({
    _appId: model.config.appId
  });
}
async function renderCommand(command, { biLogger, errorReporter, panorama }, cliVersion, tree) {
  await runCommand(
    command,
    { biLogger, errorReporter, panorama },
    cliVersion,
    async (model) => {
      const flow = getFlow(model);
      biLogger.updateDefaults({ command: command.name(), flow });
      errorReporter.setTag("flow", flow);
      const debugLogFilePath = getDebugLogFilePath(model.projectFolder);
      const handleLogin = (userInfo) => {
        errorReporter.setUser({ id: userInfo.userId });
        addUserFields(biLogger, userInfo.userId);
      };
      const headers = buildHttpClientHeaders({
        flow,
        command: command.name(),
        cliVersion
      });
      await render(
        /* @__PURE__ */ import_react.default.createElement(BiProvider, { value: biLogger }, /* @__PURE__ */ import_react.default.createElement(ErrorReporterProvider, { value: errorReporter }, /* @__PURE__ */ import_react.default.createElement(PanoramaProvider, { value: panorama }, /* @__PURE__ */ import_react.default.createElement(AccountAuthProvider, { onLogin: handleLogin }, /* @__PURE__ */ import_react.default.createElement(HttpClientProvider, { headers }, /* @__PURE__ */ import_react.default.createElement(DebugLogProvider, { filePath: debugLogFilePath }, /* @__PURE__ */ import_react.default.createElement(ProjectModelProvider, { value: model }, tree)))))))
      );
    }
  );
}
async function runCommand(command, services, cliVersion, commandFn) {
  const model = await loadProjectModel(cwd());
  await initServices(model, services);
  services.biLogger.updateDefaults({
    command: command.name()
  });
  const commandStatus = reportCommandStartEvent({
    biLogger: services.biLogger,
    command,
    cliVersion,
    flow: getFlow(model),
    extensions: {},
    extraFields: void 0
  });
  try {
    await commandFn(model);
    commandStatus.succeed();
  } catch (e) {
    commandStatus.failed(e);
    throw e;
  }
}

export {
  renderCommand,
  runCommand
};
