// @wix/cli/build/chunk-OEC6VWWW.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  GIT_IGNORED_DIR,
  useProjectModel
} from "./chunk-OPNNOJNG.js";
import {
  useHttpClient
} from "./chunk-RHVTQ2DQ.js";
import {
  usePanoramaMethodLogger
} from "./chunk-RDT7NRVZ.js";
import {
  SiteAuthProvider,
  isHttpApplicationError,
  isHttpError,
  pRetry,
  resolveUrl,
  serializer,
  toURLSearchParams
} from "./chunk-3Q7XZST5.js";
import {
  Alert,
  Spinner,
  useAsync
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import {
  getTestOverrides
} from "./chunk-WDXANXXO.js";
import {
  outputDir,
  outputFile,
  removeDir
} from "./chunk-2OYCKIQP.js";
import {
  CliError,
  CliErrorCode,
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/TranslationPullCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/TranslationPullCommand/TranslationPullCommand.tsx
init_esm_shims();
var import_react3 = __toESM(require_react(), 1);
var import_variant2 = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/TranslationPullCommand/TranslationPull.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);

// ../multilingual-client/src/index.ts
init_esm_shims();

// ../multilingual-client/src/multilingual-client.ts
init_esm_shims();

// ../../node_modules/@wix/ambassador-multilingual-localization-public-v3-published-content/build/es/http.impl.js
init_esm_shims();
var _appEmbedData = { image: "_media" };
var _audioData = { audio: "_media", coverImage: "_media" };
var _background = { image: "_media" };
var _backgroundImage = { media: "_media" };
var _design = { poll: "_pollDesign" };
var _document = {
  urlExpirationDate: "google.protobuf.Timestamp",
  thumbnail: "_image"
};
var _focalPoint = { x: "DOUBLE", y: "DOUBLE" };
var _galleryData = { items: "_item", options: "_galleryOptions" };
var _galleryOptions = { item: "_itemStyle" };
var _image = {
  urlExpirationDate: "google.protobuf.Timestamp",
  focalPoint: "_focalPoint"
};
var _imageData = { image: "_media" };
var _item = { image: "_itemImage", video: "_video" };
var _itemImage = { media: "_media" };
var _itemStyle = { ratio: "google.protobuf.DoubleValue" };
var _layoutData = {
  backgroundImage: "_backgroundImage",
  backdropImage: "_backgroundImage"
};
var _mapData = { mapSettings: "_mapSettings" };
var _mapSettings = {
  lat: "google.protobuf.DoubleValue",
  lng: "google.protobuf.DoubleValue"
};
var _media = { duration: "google.protobuf.DoubleValue" };
var _metadata = {
  createdTimestamp: "google.protobuf.Timestamp",
  updatedTimestamp: "google.protobuf.Timestamp"
};
var _node = {
  galleryData: "_galleryData",
  imageData: "_imageData",
  mapData: "_mapData",
  pollData: "_pollData",
  appEmbedData: "_appEmbedData",
  videoData: "_videoData",
  audioData: "_audioData",
  layoutData: "_layoutData",
  nodes: "_node"
};
var _option = { image: "_media" };
var _poll = { image: "_media", options: "_option" };
var _pollData = { poll: "_poll", design: "_design" };
var _pollDesign = { background: "_background" };
var _publishedContent = { fields: "_publishedContentField" };
var _publishedContentField = {
  richContent: "_richContent",
  image: "_image",
  video: "_videoV2",
  document: "_document"
};
var _queryPublishedContentRequest = {};
var _queryPublishedContentResponse = {
  publishedContent: "_publishedContent"
};
var _richContent = { nodes: "_node", metadata: "_metadata" };
var _video = { media: "_media", thumbnail: "_media" };
var _videoData = { video: "_media", thumbnail: "_media" };
var _videoResolution = {
  urlExpirationDate: "google.protobuf.Timestamp",
  poster: "_image"
};
var _videoV2 = {
  urlExpirationDate: "google.protobuf.Timestamp",
  resolutions: "_videoResolution",
  posters: "_image"
};
function resolveComWixpressLocalizationpublicV3TranslationPublishedContentUrl(opts) {
  var domainToMappings = {
    _: [
      {
        srcPath: "/localization-public",
        destPath: ""
      },
      {
        srcPath: "/translation-published-content/v3",
        destPath: "/v3"
      },
      {
        srcPath: "/_api/translation-published-content/v3",
        destPath: "/v3"
      }
    ],
    "cashier.wixapps.net": [
      {
        srcPath: "/multilingual/v1/public-localized-content",
        destPath: "/v1/public-localized-content"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/translation-published-content/v3",
        destPath: "/v3"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/translation-published-content/v3",
        destPath: "/v3"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/translation-published-content/v3",
        destPath: "/v3"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/translation-published-content/v3",
        destPath: "/v3"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/translation-published-content/v3",
        destPath: "/v3"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/translation-published-content/v3",
        destPath: "/v3"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function queryPublishedContent(payload) {
  var _a = serializer(_queryPublishedContentRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_queryPublishedContentResponse, {
    _appEmbedData,
    _audioData,
    _background,
    _backgroundImage,
    _design,
    _document,
    _focalPoint,
    _galleryData,
    _galleryOptions,
    _image,
    _imageData,
    _item,
    _itemImage,
    _itemStyle,
    _layoutData,
    _mapData,
    _mapSettings,
    _media,
    _metadata,
    _node,
    _option,
    _poll,
    _pollData,
    _pollDesign,
    _publishedContent,
    _publishedContentField,
    _richContent,
    _video,
    _videoData,
    _videoResolution,
    _videoV2
  }).fromJSON;
  function __queryPublishedContent(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.multilingual.localization_public.v3.published_content",
      method: "GET",
      methodFqn: "com.wixpress.localizationpublic.v3.TranslationPublishedContent.QueryPublishedContent",
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressLocalizationpublicV3TranslationPublishedContentUrl({
        protoPath: "/v3/published-contents/query",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData, true),
      transformResponse: fromRes,
      fallback: [
        {
          method: "POST",
          url: resolveComWixpressLocalizationpublicV3TranslationPublishedContentUrl({
            protoPath: "/v3/published-contents/query",
            data: serializedData,
            host
          }),
          data: serializedData
        }
      ]
    };
    return metadata;
  }
  __queryPublishedContent.fromReq = fromReq;
  __queryPublishedContent.__isAmbassador = true;
  return __queryPublishedContent;
}

// ../../node_modules/@wix/ambassador-multilingual-localization-public-v3-published-content/build/es/types.impl.js
init_esm_shims();
var SchemaScope;
(function(SchemaScope2) {
  SchemaScope2["UNKNOWN_SCOPE"] = "UNKNOWN_SCOPE";
  SchemaScope2["GLOBAL"] = "GLOBAL";
  SchemaScope2["SITE"] = "SITE";
})(SchemaScope || (SchemaScope = {}));
var NodeType;
(function(NodeType2) {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
})(NodeType || (NodeType = {}));
var WidthType;
(function(WidthType2) {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
})(WidthType || (WidthType = {}));
var PluginContainerDataAlignment;
(function(PluginContainerDataAlignment2) {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
})(PluginContainerDataAlignment || (PluginContainerDataAlignment = {}));
var Type;
(function(Type2) {
  Type2["LINK"] = "LINK";
  Type2["ACTION"] = "ACTION";
})(Type || (Type = {}));
var Target;
(function(Target2) {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
})(Target || (Target = {}));
var TextAlignment;
(function(TextAlignment2) {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
})(TextAlignment || (TextAlignment = {}));
var LineStyle;
(function(LineStyle2) {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
})(LineStyle || (LineStyle = {}));
var Width;
(function(Width2) {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
})(Width || (Width = {}));
var Alignment;
(function(Alignment2) {
  Alignment2["CENTER"] = "CENTER";
  Alignment2["LEFT"] = "LEFT";
  Alignment2["RIGHT"] = "RIGHT";
})(Alignment || (Alignment = {}));
var ViewMode;
(function(ViewMode2) {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
})(ViewMode || (ViewMode = {}));
var LayoutType;
(function(LayoutType2) {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
})(LayoutType || (LayoutType = {}));
var Orientation;
(function(Orientation2) {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
})(Orientation || (Orientation = {}));
var Crop;
(function(Crop2) {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
})(Crop || (Crop = {}));
var ThumbnailsAlignment;
(function(ThumbnailsAlignment2) {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
})(ThumbnailsAlignment || (ThumbnailsAlignment = {}));
var GIFType;
(function(GIFType2) {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
})(GIFType || (GIFType = {}));
var Source;
(function(Source2) {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  Source2["AI"] = "AI";
})(Source || (Source = {}));
var Position;
(function(Position2) {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  Position2["HIDDEN"] = "HIDDEN";
})(Position || (Position = {}));
var MapType;
(function(MapType2) {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
})(MapType || (MapType = {}));
var ViewRole;
(function(ViewRole2) {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
})(ViewRole || (ViewRole = {}));
var VoteRole;
(function(VoteRole2) {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
})(VoteRole || (VoteRole = {}));
var PollLayoutType;
(function(PollLayoutType2) {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
})(PollLayoutType || (PollLayoutType = {}));
var PollLayoutDirection;
(function(PollLayoutDirection2) {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
})(PollLayoutDirection || (PollLayoutDirection = {}));
var BackgroundType;
(function(BackgroundType2) {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
})(BackgroundType || (BackgroundType = {}));
var DecorationType;
(function(DecorationType2) {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  DecorationType2["FONT_FAMILY"] = "FONT_FAMILY";
})(DecorationType || (DecorationType = {}));
var FontType;
(function(FontType2) {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
})(FontType || (FontType = {}));
var ImageStylesPosition;
(function(ImageStylesPosition2) {
  ImageStylesPosition2["START"] = "START";
  ImageStylesPosition2["END"] = "END";
  ImageStylesPosition2["TOP"] = "TOP";
})(ImageStylesPosition || (ImageStylesPosition = {}));
var AspectRatio;
(function(AspectRatio2) {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
})(AspectRatio || (AspectRatio = {}));
var Resizing;
(function(Resizing2) {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
})(Resizing || (Resizing = {}));
var Placement;
(function(Placement2) {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
})(Placement || (Placement = {}));
var CardStylesType;
(function(CardStylesType2) {
  CardStylesType2["CONTAINED"] = "CONTAINED";
  CardStylesType2["FRAMELESS"] = "FRAMELESS";
})(CardStylesType || (CardStylesType = {}));
var CardStylesAlignment;
(function(CardStylesAlignment2) {
  CardStylesAlignment2["START"] = "START";
  CardStylesAlignment2["CENTER"] = "CENTER";
  CardStylesAlignment2["END"] = "END";
})(CardStylesAlignment || (CardStylesAlignment = {}));
var CardStylesLayout;
(function(CardStylesLayout2) {
  CardStylesLayout2["STACKED"] = "STACKED";
  CardStylesLayout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
})(CardStylesLayout || (CardStylesLayout = {}));
var AppType;
(function(AppType2) {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
})(AppType || (AppType = {}));
var InitialExpandedItems;
(function(InitialExpandedItems2) {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
})(InitialExpandedItems || (InitialExpandedItems = {}));
var Direction;
(function(Direction2) {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
})(Direction || (Direction = {}));
var VerticalAlignment;
(function(VerticalAlignment2) {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
})(VerticalAlignment || (VerticalAlignment = {}));
var NullValue;
(function(NullValue2) {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
})(NullValue || (NullValue = {}));
var Scaling;
(function(Scaling2) {
  Scaling2["AUTO"] = "AUTO";
  Scaling2["CONTAIN"] = "CONTAIN";
  Scaling2["COVER"] = "COVER";
})(Scaling || (Scaling = {}));
var ImagePosition;
(function(ImagePosition2) {
  ImagePosition2["CENTER"] = "CENTER";
  ImagePosition2["CENTER_LEFT"] = "CENTER_LEFT";
  ImagePosition2["CENTER_RIGHT"] = "CENTER_RIGHT";
  ImagePosition2["TOP"] = "TOP";
  ImagePosition2["TOP_LEFT"] = "TOP_LEFT";
  ImagePosition2["TOP_RIGHT"] = "TOP_RIGHT";
  ImagePosition2["BOTTOM"] = "BOTTOM";
  ImagePosition2["BOTTOM_LEFT"] = "BOTTOM_LEFT";
  ImagePosition2["BOTTOM_RIGHT"] = "BOTTOM_RIGHT";
})(ImagePosition || (ImagePosition = {}));
var VerticalAlignmentAlignment;
(function(VerticalAlignmentAlignment2) {
  VerticalAlignmentAlignment2["TOP"] = "TOP";
  VerticalAlignmentAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignmentAlignment2["BOTTOM"] = "BOTTOM";
})(VerticalAlignmentAlignment || (VerticalAlignmentAlignment = {}));
var ResponsivenessBehaviour;
(function(ResponsivenessBehaviour2) {
  ResponsivenessBehaviour2["STACK"] = "STACK";
  ResponsivenessBehaviour2["WRAP"] = "WRAP";
})(ResponsivenessBehaviour || (ResponsivenessBehaviour = {}));
var DesignTarget;
(function(DesignTarget2) {
  DesignTarget2["LAYOUT"] = "LAYOUT";
  DesignTarget2["CELL"] = "CELL";
})(DesignTarget || (DesignTarget = {}));
var SortOrder;
(function(SortOrder2) {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
var WebhookIdentityType;
(function(WebhookIdentityType2) {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));

// ../../node_modules/@wix/ambassador-multilingual-locales-v2-locale/build/es/http.impl.js
init_esm_shims();
var _locale = {
  createdDate: "google.protobuf.Timestamp",
  updatedDate: "google.protobuf.Timestamp"
};
var _queryLocalesRequest = {};
var _queryLocalesResponse = { locales: "_locale" };
function resolveWixMultilingualLocalesV2LocalesServiceUrl(opts) {
  var domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/locales",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/locales",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/locales",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/locales",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/locales",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/locales",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/locales",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function queryLocales(payload) {
  var _a = serializer(_queryLocalesRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_queryLocalesResponse, { _locale }).fromJSON;
  function __queryLocales(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.multilingual.locales.v2.locale",
      method: "GET",
      methodFqn: "wix.multilingual.locales.v2.LocalesService.QueryLocales",
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveWixMultilingualLocalesV2LocalesServiceUrl({
        protoPath: "/v2/locale/query",
        data: serializedData,
        host
      }),
      params: toURLSearchParams(serializedData, true),
      transformResponse: fromRes,
      fallback: [
        {
          method: "POST",
          url: resolveWixMultilingualLocalesV2LocalesServiceUrl({
            protoPath: "/v2/locale/query",
            data: serializedData,
            host
          }),
          data: serializedData
        }
      ]
    };
    return metadata;
  }
  __queryLocales.fromReq = fromReq;
  __queryLocales.__isAmbassador = true;
  return __queryLocales;
}

// ../multilingual-client/src/multilingual-client.ts
var HEADLESS_APP_ID = "78640cbb-be47-45f8-b6de-8ed97287872c";
var HEADLESS_TRANSLATIONS_ENTITY_TYPE = "headless-site";
var MultilingualClient = class {
  constructor(httpClient) {
    this.httpClient = httpClient;
    const { minRetryTimeout } = getTestOverrides();
    this.retryOptions = {
      retries: 3,
      minTimeout: minRetryTimeout,
      maxTimeout: 3 * 1e3
    };
  }
  retryOptions;
  queryPublishedContents = async (request) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          queryPublishedContent({
            ...request
          })
        ),
        this.retryOptions
      );
      return data;
    } catch (error) {
      if (isHttpError(error) && isHttpApplicationError(error)) {
        this.handleApplicationError(error);
      }
      throw error;
    }
  };
  queryAllPublishedContents = async (request = { query: {} }) => {
    const allContents = [];
    let cursor;
    const limit = 100;
    let hasMore = true;
    while (hasMore) {
      let sort = request.query.sort ?? [
        {
          fieldName: "id",
          order: SortOrder.ASC
        }
      ];
      let filter = {
        ...request.query.filter ?? {},
        "schemaKey.appId": HEADLESS_APP_ID,
        "schemaKey.scope": SchemaScope.GLOBAL,
        "schemaKey.entityType": HEADLESS_TRANSLATIONS_ENTITY_TYPE
      };
      if (cursor) {
        sort = void 0;
        filter = void 0;
      }
      const response = await this.queryPublishedContents({
        query: {
          filter,
          cursorPaging: {
            limit,
            ...cursor ? { cursor } : {}
          },
          sort
        }
      });
      const contents = response.publishedContent ?? [];
      allContents.push(...contents);
      const pagingMetadata = response.pagingMetadata;
      hasMore = pagingMetadata?.hasNext ?? false;
      cursor = pagingMetadata?.cursors?.next ?? void 0;
      if (hasMore && !cursor) {
        hasMore = false;
      }
    }
    return allContents;
  };
  queryLocales = async (request) => {
    try {
      const { data } = await pRetry(
        () => this.httpClient.request(
          queryLocales({
            ...request
          })
        ),
        this.retryOptions
      );
      return data;
    } catch (error) {
      if (isHttpError(error) && isHttpApplicationError(error)) {
        throw new CliError({
          code: CliErrorCode.FailedToQueryLocales(),
          cause: error
        });
      }
      throw error;
    }
  };
  getPrimaryLanguage = async () => {
    const response = await this.queryLocales();
    const primaryLocale = response.locales?.find(
      (locale) => locale.visitorPrimaryLocale === true
    );
    return primaryLocale?.languageCode ?? void 0;
  };
  handleApplicationError(error) {
    const applicationError = error.response?.data.details.applicationError;
    if (applicationError?.code === "APP_NOT_INSTALLED" && error.response?.status === 428) {
      throw new CliError({
        code: CliErrorCode.MultilingualAppNotInstalled(),
        cause: error
      });
    }
    throw new CliError({
      code: CliErrorCode.FailedToQueryPublishedContents(),
      cause: error
    });
  }
};

// ../multilingual-client/src/useMultilingualClient.ts
init_esm_shims();
var import_react = __toESM(require_react(), 1);
function useMultilingualClient() {
  const httpClient = useHttpClient({ type: "api" });
  const client = (0, import_react.useMemo)(
    () => new MultilingualClient(httpClient),
    [httpClient]
  );
  return usePanoramaMethodLogger(client);
}

// ../cli-astro-commands/src/components/TranslationPullCommand/TranslationPull.tsx
import { join as join2 } from "node:path";

// ../cli-astro-commands/src/components/TranslationPullCommand/save-translation-files.ts
init_esm_shims();
import { join } from "node:path";
async function saveTranslationFiles(translationsDir, multilingualDir, contents, primaryLanguage) {
  await removeDir(multilingualDir);
  await outputDir(translationsDir);
  if (!contents.length) {
    throw new CliError({
      code: CliErrorCode.NoTranslationPublishedContents(),
      cause: null
    });
  }
  const translationsByLocale = /* @__PURE__ */ new Map();
  for (const content of contents) {
    const locale = content.locale ?? "";
    if (!translationsByLocale.has(locale)) {
      translationsByLocale.set(locale, []);
    }
    translationsByLocale.get(locale).push(content);
  }
  for (const [locale, localeContents] of translationsByLocale) {
    const fileName = `${locale}.json`;
    const filePath = join(translationsDir, fileName);
    const translations = {};
    for (const content of localeContents) {
      const firstField = content.fields?.[0] ?? {};
      if (firstField.textValue && content.entityId) {
        translations[content.entityId] = firstField.textValue;
      }
    }
    await outputFile(filePath, JSON.stringify(translations, null, 2));
  }
  const metadataPath = join(multilingualDir, "metadata.json");
  await outputFile(
    metadataPath,
    JSON.stringify(
      {
        totalItems: contents.length,
        locales: Array.from(translationsByLocale.keys()),
        primaryLanguage,
        itemsByLocale: Object.fromEntries(
          Array.from(translationsByLocale.entries()).map(([locale, items]) => [
            locale,
            items.length
          ])
        )
      },
      null,
      2
    )
  );
}

// ../cli-astro-commands/src/components/TranslationPullCommand/TranslationPull.tsx
var TranslationPull = ({
  onComplete
}) => {
  const { model } = useProjectModel();
  const multilingualClient = useMultilingualClient();
  const { status } = useAsync(async () => {
    const [allContents, primaryLanguage] = await Promise.all([
      multilingualClient.queryAllPublishedContents(),
      multilingualClient.getPrimaryLanguage()
    ]);
    const multilingualDir = join2(
      model.projectFolder,
      GIT_IGNORED_DIR,
      "multilingual"
    );
    const translationsDir = join2(multilingualDir, "translations");
    await saveTranslationFiles(
      translationsDir,
      multilingualDir,
      allContents,
      primaryLanguage
    );
    onComplete?.();
  }, []);
  return (0, import_variant.match)(status, {
    Loading: () => /* @__PURE__ */ import_react2.default.createElement(Spinner, { text: "Pulling translation files..." }),
    Success: () => {
      return /* @__PURE__ */ import_react2.default.createElement(Alert, { type: "success" }, 'Translation files written to "', join2(GIT_IGNORED_DIR, "multilingual/translations"), '"');
    },
    Error: () => null,
    NotRequested: () => null
  });
};

// ../cli-astro-commands/src/components/TranslationPullCommand/TranslationPullCommand.tsx
var TranslationPullCommand = ({
  onComplete
}) => {
  const {
    model: { config }
  } = useProjectModel();
  const { status } = useAsync(async () => {
    if (config.projectType !== "Site") {
      throw new CliError({
        code: CliErrorCode.TranslationUnsupportedForProjectType(),
        cause: null
      });
    }
    return { siteId: config.siteId };
  }, []);
  return (0, import_variant2.match)(status, {
    Loading: () => null,
    Error: () => null,
    Success: ({ result: { siteId } }) => {
      return /* @__PURE__ */ import_react3.default.createElement(SiteAuthProvider, { siteId }, /* @__PURE__ */ import_react3.default.createElement(TranslationPull, { onComplete }));
    }
  });
};

export {
  TranslationPullCommand
};
