// @wix/cli/build/chunk-5AY2TF4R.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  DeploymentPipelinesStatus
} from "./chunk-CH52GF36.js";
import {
  external_exports
} from "./chunk-F43XHKVL.js";
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site-old/src/schemas.ts
init_esm_shims();
var deployAppResponseSchema = external_exports.object({
  deploymentInfo: external_exports.object({
    deploymentId: external_exports.string(),
    deploymentUrl: external_exports.string().url(),
    deploymentShortUrl: external_exports.string().nullish(),
    deployedRevision: external_exports.string(),
    isPublishPipelineDeployment: external_exports.boolean()
  })
});
var resolveNpmDependenciesSchema = external_exports.object({
  jobId: external_exports.string().uuid()
});
var npmDependenciesLockFileResponseSchema = external_exports.object({
  result: external_exports.object({
    npmDependenciesLockFile: external_exports.string().url()
  })
});
var getDeploymentResponseSchema = external_exports.object({
  deployment: external_exports.object({
    deploymentPipelinesStatus: external_exports.enum([
      DeploymentPipelinesStatus.IN_PROGRESS,
      DeploymentPipelinesStatus.SUCCESS,
      DeploymentPipelinesStatus.ERROR
    ])
  })
});

export {
  deployAppResponseSchema,
  resolveNpmDependenciesSchema,
  npmDependenciesLockFileResponseSchema,
  getDeploymentResponseSchema
};
