// @wix/cli/build/chunk-FRO7EYZX.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import {
  external_exports
} from "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/GenerateCommand/generateFlows.ts
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var DevCenterGenerateFlow = (0, import_variant.variant)({
  DashboardPage: {},
  DashboardPlugin: {},
  DashboardMenuPlugin: {},
  DashboardModal: {},
  Event: {},
  ServicePlugin: {},
  EmbeddedScript: {},
  CustomElement: {},
  SitePlugin: {}
});
var getGenerateFlowByKey = (key) => {
  return (0, import_variant.match)(key, {
    DashboardPage: () => ({
      key,
      title: "Dashboard Page",
      value: DevCenterGenerateFlow.DashboardPage()
    }),
    DashboardModal: () => ({
      key,
      title: "Dashboard Modal",
      value: DevCenterGenerateFlow.DashboardModal()
    }),
    DashboardPlugin: () => ({
      key,
      title: "Dashboard Plugin",
      value: DevCenterGenerateFlow.DashboardPlugin()
    }),
    DashboardMenuPlugin: () => ({
      key,
      title: "Dashboard Menu Plugin",
      value: DevCenterGenerateFlow.DashboardMenuPlugin()
    }),
    CustomElement: () => ({
      key,
      title: "Custom Element",
      value: DevCenterGenerateFlow.CustomElement()
    }),
    EmbeddedScript: () => ({
      key,
      title: "Embedded Script",
      value: DevCenterGenerateFlow.EmbeddedScript()
    }),
    SitePlugin: () => ({
      key,
      title: "Site Plugin",
      value: DevCenterGenerateFlow.SitePlugin()
    }),
    Event: () => ({
      key,
      title: "Event",
      value: DevCenterGenerateFlow.Event()
    }),
    ServicePlugin: () => ({
      key,
      title: "Service Plugin",
      value: DevCenterGenerateFlow.ServicePlugin()
    })
  });
};
var useGenerateFlows = () => {
  return (0, import_react.useMemo)(() => {
    return [
      {
        title: "Dashboard",
        items: [
          getGenerateFlowByKey("DashboardPage"),
          getGenerateFlowByKey("DashboardModal"),
          getGenerateFlowByKey("DashboardPlugin"),
          getGenerateFlowByKey("DashboardMenuPlugin")
        ]
      },
      {
        title: "Site",
        items: [
          getGenerateFlowByKey("EmbeddedScript"),
          getGenerateFlowByKey("CustomElement"),
          getGenerateFlowByKey("SitePlugin")
        ]
      },
      {
        title: "Backend",
        items: [
          getGenerateFlowByKey("Event"),
          getGenerateFlowByKey("ServicePlugin")
        ]
      }
    ];
  }, []);
};

// ../cli-astro-commands/src/components/GenerateCommand/docsExtensionType.ts
init_esm_shims();
var import_variant2 = __toESM(require_lib(), 1);
var docsExtensionTypeSchema = external_exports.enum([
  "DASHBOARD_PAGE",
  "DASHBOARD_MODAL",
  "DASHBOARD_PLUGIN",
  "DASHBOARD_MENU_PLUGIN",
  "EMBEDDED_SCRIPT",
  "CUSTOM_ELEMENT",
  "SITE_PLUGIN",
  "EVENT",
  "SERVICE_PLUGIN"
]);
var getGenerateFlowByDocsExtensionType = (type) => {
  return (0, import_variant2.match)(type, {
    DASHBOARD_PAGE: () => getGenerateFlowByKey("DashboardPage"),
    DASHBOARD_MODAL: () => getGenerateFlowByKey("DashboardModal"),
    DASHBOARD_PLUGIN: () => getGenerateFlowByKey("DashboardPlugin"),
    DASHBOARD_MENU_PLUGIN: () => getGenerateFlowByKey("DashboardMenuPlugin"),
    EMBEDDED_SCRIPT: () => getGenerateFlowByKey("EmbeddedScript"),
    CUSTOM_ELEMENT: () => getGenerateFlowByKey("CustomElement"),
    SITE_PLUGIN: () => getGenerateFlowByKey("SitePlugin"),
    EVENT: () => getGenerateFlowByKey("Event"),
    SERVICE_PLUGIN: () => getGenerateFlowByKey("ServicePlugin")
  });
};

export {
  useGenerateFlows,
  docsExtensionTypeSchema,
  getGenerateFlowByDocsExtensionType
};
