// @wix/cli/build/DevCommand-N5CAC2JX.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useTunnel
} from "./chunk-DFA5L4HJ.js";
import {
  useWixCliAppBi
} from "./chunk-CW6TLWH7.js";
import {
  PlatformLink,
  RedirectLink,
  SiteProvider,
  SiteSelector,
  createRedirectLink,
  readAppConfig,
  saveAppConfig,
  useARMClient,
  useSite,
  useUpdateDeploymentTopology
} from "./chunk-CZ76ZPHP.js";
import "./chunk-42GS4QEG.js";
import "./chunk-SC2W3WT3.js";
import {
  createPackageManager,
  getRepoType
} from "./chunk-AT5WRPKM.js";
import {
  execa
} from "./chunk-2R5BII2J.js";
import "./chunk-W54PG6O5.js";
import "./chunk-YLWTNITT.js";
import {
  ComponentType,
  devMetadataSchema,
  looseComponentSchema,
  parseLooseAppManifest,
  useProjectModel
} from "./chunk-OPNNOJNG.js";
import {
  useHttpClient
} from "./chunk-RHVTQ2DQ.js";
import {
  usePanoramaMethodLogger
} from "./chunk-RDT7NRVZ.js";
import {
  ErrorViewer,
  SiteAuthProvider,
  resolveUrl,
  serializer,
  useBiLogger,
  useDebugLog,
  useErrorReporter
} from "./chunk-3Q7XZST5.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-6WEX76B7.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Box_default,
  ConfirmInput,
  Key,
  PointerIcon,
  Spinner,
  Text,
  createRendererToString,
  useAsync,
  useAsyncCallback,
  useExit,
  use_input_default
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import {
  stripAnsi
} from "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import {
  readUserConfig,
  updateUserConfig
} from "./chunk-TKJGN2XT.js";
import "./chunk-C5PTSUGF.js";
import {
  getTestOverrides
} from "./chunk-WDXANXXO.js";
import "./chunk-2OYCKIQP.js";
import {
  external_exports
} from "./chunk-F43XHKVL.js";
import {
  CliError,
  CliErrorCode,
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/DevCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/DevCommand/DevCommand.tsx
init_esm_shims();
var import_react15 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/DevCommand/utils/runAstroDev.ts
init_esm_shims();
var EVENT_LINE = "line";
var EVENT_SERVER_ADDRESS = "server_address";
function runAstroDev({
  packageManager,
  projectFolder,
  port,
  allowedHosts
}) {
  const binaryArgs = ["astro", "dev"];
  if (port) {
    binaryArgs.push("--port", `${port}`);
  }
  if (allowedHosts) {
    binaryArgs.push("--allowed-hosts", allowedHosts);
  }
  const { file, args } = packageManager.getRunBinaryCmd(binaryArgs);
  const result = execa(file, args, {
    cwd: projectFolder,
    stdio: "pipe",
    env: { FORCE_COLOR: "1" },
    cleanup: true,
    windowsHide: false
  });
  let stdoutBuffer = "";
  result.stdout?.on("data", (chunk) => {
    stdoutBuffer += chunk.toString();
    const lines = stdoutBuffer.split("\n");
    stdoutBuffer = lines.pop() ?? "";
    for (const line of lines) {
      result.stdout?.emit(EVENT_LINE, line);
      if (line.includes("http://")) {
        const match5 = /(http:\/\/[^\s]+)/.exec(line);
        if (match5?.[1]) {
          const address = stripAnsi(match5[1]).trim();
          result.stdout?.emit(EVENT_SERVER_ADDRESS, address);
        }
      }
    }
  });
  let stdErrBuffer = "";
  result.stderr?.on("data", (chunk) => {
    stdErrBuffer += chunk.toString();
    const lines = stdErrBuffer.split("\n");
    stdErrBuffer = lines.pop() ?? "";
    for (const line of lines) {
      result.stderr?.emit(EVENT_LINE, line);
    }
  });
  return result;
}

// ../cli-astro-commands/src/components/DevCommand/utils/createAstroLogger.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var dateTimeFormat = new Intl.DateTimeFormat([], {
  hour: "2-digit",
  minute: "2-digit",
  second: "2-digit",
  hour12: false
});
function createAstroLogger(isPrefixed = true) {
  const renderer = createRendererToString({ stripAnsi: false });
  const log = (msg) => {
    if (typeof msg === "string") {
      setTimeout(() => console.log(msg), 0);
    } else {
      setTimeout(() => console.log(renderer.render(msg)), 0);
    }
  };
  const prefixedLog = (msg) => {
    const timestamp = dateTimeFormat.format(/* @__PURE__ */ new Date());
    const log2 = /* @__PURE__ */ import_react.default.createElement(Text, null, /* @__PURE__ */ import_react.default.createElement(Text, null, timestamp, " "), /* @__PURE__ */ import_react.default.createElement(Text, { skin: "info" }, "[wix] "), /* @__PURE__ */ import_react.default.createElement(Text, null, msg));
    console.log(renderer.render(log2));
  };
  return isPrefixed ? { log: prefixedLog } : { log };
}

// ../cli-astro-commands/src/components/DevCommand/hooks/useFetchDevMetadata.ts
init_esm_shims();
function useFetchDevMetadata() {
  const http = useHttpClient({ type: "standalone" });
  return useAsyncCallback(async (_, serverLocalUrl) => {
    const url = new URL("/_wix/dev-metadata.json", serverLocalUrl);
    try {
      const { data } = await http.get(url.toString());
      return {
        devMetadata: devMetadataSchema.parse(data),
        serverLocalUrl
      };
    } catch {
      return {
        devMetadata: null,
        serverLocalUrl
      };
    }
  });
}

// ../cli-astro-commands/src/components/DevCommand/SiteDevFlow.tsx
init_esm_shims();
var import_react8 = __toESM(require_react(), 1);
var import_variant4 = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/DevCommand/DevScreenBorder.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);
var DevScreenBorder = ({ children }) => {
  return /* @__PURE__ */ import_react2.default.createElement(
    Box_default,
    {
      flexDirection: "column",
      borderTop: true,
      borderLeft: false,
      borderBottom: false,
      borderRight: false,
      borderStyle: "round",
      borderColor: "blueBright",
      width: "95%"
    },
    children
  );
};

// ../cli-astro-commands/src/site/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/DevCommand/SiteDev.tsx
init_esm_shims();
var import_react7 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/DevCommand/hooks/useDevCenterUpdater.ts
init_esm_shims();
var import_react5 = __toESM(require_react(), 1);
var import_variant2 = __toESM(require_lib(), 1);

// ../metasite-manager-client/src/index.ts
init_esm_shims();

// ../metasite-manager-client/src/useMetasiteManagerClient.ts
init_esm_shims();
var import_react3 = __toESM(require_react(), 1);

// ../metasite-manager-client/src/metasite-manager-client.ts
init_esm_shims();

// ../../node_modules/@wix/ambassador-metasite-api-v1-meta-site/build/es/http.impl.js
init_esm_shims();
var _createMetaSiteFromTemplateRequest = {};
var _createMetaSiteFromTemplateResponse = { metaSite: "_metaSite" };
var _metaSite = {
  dateCreated: "google.protobuf.Timestamp",
  dateUpdated: "google.protobuf.Timestamp",
  accountCreatedDate: "google.protobuf.Timestamp"
};
function resolveComWixpressMetasiteApiV1MetaSiteLifecycleUrl(opts) {
  var domainToMappings = {
    "www._base_domain_": [
      {
        srcPath: "/_api/msm-writer",
        destPath: ""
      },
      {
        srcPath: "/_api/msm",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/msm-writer",
        destPath: ""
      },
      {
        srcPath: "/_api/msm",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/msm",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/msm-writer",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/msm-writer",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/msm-writer",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function createMetaSiteFromTemplate(payload) {
  var _a = serializer(_createMetaSiteFromTemplateRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_createMetaSiteFromTemplateResponse, { _metaSite }).fromJSON;
  function __createMetaSiteFromTemplate(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.metasite.api.v1.meta_site",
      method: "POST",
      methodFqn: "com.wixpress.metasite.api.v1.MetaSiteLifecycle.CreateMetaSiteFromTemplate",
      url: resolveComWixpressMetasiteApiV1MetaSiteLifecycleUrl({
        protoPath: "/v1/meta-site/create-from-template",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __createMetaSiteFromTemplate.fromReq = fromReq;
  __createMetaSiteFromTemplate.__isAmbassador = true;
  return __createMetaSiteFromTemplate;
}

// ../../node_modules/@wix/ambassador-document-management-document-store-v1-transaction/build/es/http.impl.js
init_esm_shims();
var _distributeRequest = {};
var _distributeResponse = {};
function resolveComWixpressEditorApiDocumentStoreServiceUrl(opts) {
  var domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/editor-document-store",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/editor-document-store",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/editor-document-store",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/editor-document-store",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/editor-document-store",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
function distribute(payload) {
  var _a = serializer(_distributeRequest, {}), toReq = _a.toJSON, fromReq = _a.fromJSON;
  var fromRes = serializer(_distributeResponse, {}).fromJSON;
  function __distribute(_a2) {
    var host = _a2.host;
    var serializedData = toReq(payload);
    var metadata = {
      entityFqdn: "wix.document_management.document_store.v1.transaction",
      method: "POST",
      methodFqn: "com.wixpress.editor.api.DocumentStoreService.Distribute",
      migrationOptions: {
        optInTransformResponse: true
      },
      url: resolveComWixpressEditorApiDocumentStoreServiceUrl({
        protoPath: "/v1/distribute",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: fromRes
    };
    return metadata;
  }
  __distribute.fromReq = fromReq;
  __distribute.__isAmbassador = true;
  return __distribute;
}

// ../metasite-manager-client/src/schemas.ts
init_esm_shims();
var clientSpecMapSchema = external_exports.record(
  external_exports.object({
    appDefinitionId: external_exports.string().optional(),
    permissions: external_exports.object({ revoked: external_exports.boolean() }).optional(),
    instanceId: external_exports.string().optional()
  })
);
var createMetaSiteFromTemplateSchema = external_exports.object({
  metaSite: external_exports.object({
    metaSiteId: external_exports.string()
  })
});

// ../metasite-manager-client/src/metasite-manager-client.ts
var MetasiteManagerClient = class {
  constructor(httpClient) {
    this.httpClient = httpClient;
  }
  getClientSpecMap = async (metaSiteId) => {
    try {
      const { data } = await this.httpClient.request({
        url: `/_api/msm/v1/meta-site/editor-client-spec-map/${metaSiteId}`
      });
      return clientSpecMapSchema.parse(data);
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetClientSpecMap(),
        cause: e
      });
    }
  };
  createMetaSiteFromTemplate = async (request) => {
    try {
      const { data } = await this.httpClient.request(
        createMetaSiteFromTemplate(request)
      );
      return createMetaSiteFromTemplateSchema.parse(data).metaSite;
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToCreateMetaSiteFromTemplate(),
        cause: e
      });
    }
  };
  getSiteInstalledApps = async (metaSiteId) => {
    try {
      const clientSpecMap = await this.getClientSpecMap(metaSiteId);
      return Object.values(clientSpecMap).filter((spec) => spec.appDefinitionId && !spec.permissions?.revoked).map((spec) => spec.appDefinitionId);
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetSiteInstalledApps(),
        cause: e
      });
    }
  };
  getInstalledAppInstanceId = async ({
    metaSiteId,
    appId
  }) => {
    try {
      const clientSpecMap = await this.getClientSpecMap(metaSiteId);
      const installedApp = Object.values(clientSpecMap).find(
        (spec) => spec.appDefinitionId === appId
      );
      return installedApp?.instanceId ?? "";
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToGetSiteInstalledAppInstanceId(),
        cause: e
      });
    }
  };
  distributeClientUpdate = async (appId) => {
    try {
      await this.httpClient.request(
        distribute({
          payloads: [
            {
              id: appId,
              messageType: "MANIFEST_UPDATE",
              data: {
                appId
              }
            }
          ],
          branchId: "00000000-0000-0000-0000-000000000000"
        })
      );
    } catch (e) {
      throw new CliError({
        code: CliErrorCode.FailedToDistributeClientManifestUpdate(),
        cause: e
      });
    }
  };
};

// ../metasite-manager-client/src/useMetasiteManagerClient.ts
function useMetasiteManagerClient() {
  const httpClient = useHttpClient({ type: "general" });
  const client = (0, import_react3.useMemo)(
    () => new MetasiteManagerClient(httpClient),
    [httpClient]
  );
  return usePanoramaMethodLogger(client);
}

// ../cli-astro-commands/src/components/DevCommand/hooks/useFetchAppManifest.ts
init_esm_shims();
var import_react4 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/DevCommand/utils/isTunnelRequired.ts
init_esm_shims();
function isTunnelRequired(appManifest) {
  for (const extension of appManifest.components) {
    const knownAppManifest = parseKnownExtensionManifest(extension);
    if (knownAppManifest != null && knownAppManifest.compType === ComponentType.WEBHOOK) {
      return !URL.canParse(knownAppManifest.compData.webhook.callbackUrl);
    }
  }
  return false;
}
function parseKnownExtensionManifest(extensionManifest) {
  const appManifestResult = looseComponentSchema.safeParse(extensionManifest);
  if (appManifestResult.error) {
    return null;
  }
  return appManifestResult.data;
}

// ../cli-astro-commands/src/components/DevCommand/hooks/useFetchAppManifest.ts
function useFetchAppManifest() {
  const http = useHttpClient({ type: "standalone" });
  return (0, import_react4.useCallback)(
    async (serverLocalUrl, devMetadata, baseUrl, tunnelUrl) => {
      const url = new URL("/_wix/app-manifest.json", serverLocalUrl);
      const { data: text } = await http.get(url.toString(), {
        responseType: "text",
        transitional: {
          silentJSONParsing: true,
          forcedJSONParsing: false,
          clarifyTimeoutError: false
        },
        disableWixHeaders: true
      });
      if (devMetadata != null) {
        const placeholders = /* @__PURE__ */ new Map();
        placeholders.set(devMetadata.staticsUrlPlaceholder, baseUrl);
        placeholders.set(devMetadata.serverUrlPlaceholder, tunnelUrl);
        return {
          manifest: parseLooseAppManifest(text, placeholders),
          isTunnelRequired: text.includes(devMetadata.serverUrlPlaceholder)
        };
      }
      const manifestV1 = parseLooseAppManifest(text);
      return {
        manifest: manifestV1,
        isTunnelRequired: isTunnelRequired(manifestV1)
      };
    },
    [http]
  );
}

// ../cli-astro-commands/src/components/DevCommand/hooks/checkManifestForChanges.ts
init_esm_shims();
var import_variant = __toESM(require_lib(), 1);
import { isDeepStrictEqual } from "node:util";
var ManifestUpdateResult = (0, import_variant.variant)({
  Changed: (0, import_variant.fields)(),
  Unchanged: {}
});
function checkManifestForChanges(previous, latest) {
  if (previous == null) {
    return ManifestUpdateResult.Changed({
      shouldDistributeClientUpdate: false
    });
  }
  const oldComponentsMap = new Map(
    previous.components.map((comp) => [comp.compId, comp])
  );
  const changedComponents = latest.components.filter((newComp) => {
    const oldComp = oldComponentsMap.get(newComp.compId);
    return !isDeepStrictEqual(newComp, oldComp);
  });
  if (changedComponents.length === 0 && previous.components.length === latest.components.length) {
    return ManifestUpdateResult.Unchanged();
  }
  const editorComponentsChanged = changedComponents.filter(
    (comp) => comp.compType === ComponentType.PANEL || comp.compType === ComponentType.EDITOR_REACT_COMPONENT
  );
  return ManifestUpdateResult.Changed({
    shouldDistributeClientUpdate: editorComponentsChanged.length > 0
  });
}

// ../cli-astro-commands/src/components/DevCommand/hooks/useDevCenterUpdater.ts
var { pollInterval } = getTestOverrides();
var useDevCenterUpdater = ({
  serverLocalUrl,
  serverExposedUrl,
  devMetadata
}) => {
  const [devArmTag, setDevArmTag] = (0, import_react5.useState)();
  const [manifest, setManifest] = (0, import_react5.useState)(null);
  const writeToDebugLog = useDebugLog();
  const { siteId } = useSite();
  const { distributeClientUpdate } = useMetasiteManagerClient();
  const {
    model: {
      config: { appId, projectType },
      projectFolder
    }
  } = useProjectModel();
  const logger = (0, import_react5.useMemo)(() => createAstroLogger(), []);
  const arm = useARMClient();
  const updateDeploymentTopology = useUpdateDeploymentTopology(siteId);
  const openTunnel = useTunnel();
  const tunnelUrl = (0, import_react5.useRef)();
  const [tunnelRequired, setTunnelRequired] = (0, import_react5.useState)(false);
  const fetchAppManifest = useFetchAppManifest();
  const updateDevCenter = (0, import_react5.useCallback)(
    async (latestManifest, isTunnelRequired2) => {
      const baseUrl = serverExposedUrl ?? serverLocalUrl;
      const manifestUpdateResult = checkManifestForChanges(
        manifest,
        latestManifest
      );
      if ((0, import_variant2.isType)(manifestUpdateResult, "Unchanged")) {
        return manifestUpdateResult;
      }
      if (!serverExposedUrl && !tunnelUrl.current && isTunnelRequired2) {
        const config = await readUserConfig();
        if (!config.tunneling) {
          setTunnelRequired(true);
          return ManifestUpdateResult.Unchanged();
        }
        const serverLocalPort = parseInt(new URL(serverLocalUrl).port, 10);
        const tunnel = await openTunnel(serverLocalPort);
        tunnelUrl.current = tunnel.url;
        return ManifestUpdateResult.Unchanged();
      }
      const appConfig = await readAppConfig(projectFolder);
      let updatedDevArmTag = appConfig?.devArmTag;
      if (!updatedDevArmTag) {
        updatedDevArmTag = await arm.createLooseAppRelease(
          latestManifest,
          appId
        );
      } else {
        const updateState = await arm.updateLooseAppRelease(
          latestManifest,
          updatedDevArmTag
        );
        if (!(0, import_variant2.isType)(updateState, "TagUpdated")) {
          updatedDevArmTag = await arm.createLooseAppRelease(
            latestManifest,
            appId
          );
        }
      }
      if (updatedDevArmTag !== appConfig?.devArmTag) {
        await saveAppConfig(projectFolder, { devArmTag: updatedDevArmTag });
      }
      if ((0, import_variant2.isType)(manifestUpdateResult, "Changed") && manifestUpdateResult.shouldDistributeClientUpdate) {
        await distributeClientUpdate(appId);
      }
      setManifest(latestManifest);
      setDevArmTag(updatedDevArmTag);
      await updateDeploymentTopology.execute({
        siteUrl: projectType === "Site" ? baseUrl : createRedirectLink({
          platform: "site",
          appId,
          tenantId: siteId,
          versionOverrideId: updatedDevArmTag
        }),
        armTag: updatedDevArmTag,
        environment: "development"
      });
      return manifestUpdateResult;
    },
    [
      siteId,
      manifest,
      distributeClientUpdate,
      serverLocalUrl,
      serverExposedUrl,
      arm,
      projectFolder,
      appId,
      updateDeploymentTopology,
      openTunnel,
      projectType
    ]
  );
  const [timer, setTimer] = (0, import_react5.useState)(Date.now());
  (0, import_react5.useEffect)(() => {
    let disposed = false;
    async function fetchAndUpdate() {
      const baseUrl = serverExposedUrl ?? serverLocalUrl;
      const { manifest: manifest2, isTunnelRequired: isTunnelRequired2 } = await fetchAppManifest(
        serverLocalUrl,
        devMetadata,
        baseUrl,
        tunnelUrl.current ?? baseUrl
      );
      try {
        const result = await updateDevCenter(manifest2, isTunnelRequired2);
        if ((0, import_variant2.isType)(result, "Changed")) {
          logger.log(
            "App Manifest updated. Refresh existing browser tabs to view the latest changes."
          );
        }
      } catch (error) {
        setManifest(manifest2);
        logger.log("Failed to update App Manifest");
        logger.log(`${error}`);
        writeToDebugLog(error);
      }
    }
    void fetchAndUpdate().finally(() => {
      if (!disposed) {
        setTimeout(() => setTimer(Date.now()), pollInterval ?? 1e3);
      }
    });
    return () => {
      disposed = true;
    };
  }, [timer]);
  return { tunnelRequired, devArmTag };
};

// ../cli-astro-commands/src/components/DevCommand/TunnelingConsent.tsx
init_esm_shims();
var import_react6 = __toESM(require_react(), 1);
var import_variant3 = __toESM(require_lib(), 1);
var TunnelingConsent = ({ required, children }) => {
  const exit = useExit();
  const [approved, setApproved] = (0, import_react6.useState)();
  const { status } = useAsync(async () => {
    const config = await readUserConfig();
    if (config.tunneling) {
      setApproved(true);
    }
    return config;
  }, []);
  if (!required || approved) {
    return children;
  }
  return /* @__PURE__ */ import_react6.default.createElement(import_react6.default.Fragment, null, (0, import_variant3.match)(status, {
    Success: () => /* @__PURE__ */ import_react6.default.createElement(
      ConfirmInput,
      {
        label: "To proceed, we'll set up a secure tunnel from your machine for development purposes",
        initialValue: true,
        onSubmit: async (approved2) => {
          setApproved(approved2);
          await updateUserConfig({ tunneling: approved2 });
          if (!approved2) {
            exit();
          }
        }
      }
    ),
    Error: () => null,
    Loading: () => null
  }), approved === false && /* @__PURE__ */ import_react6.default.createElement(Text, { skin: "error" }, "Operation aborted. Tunneling is necessary to run your application."));
};

// ../cli-astro-commands/src/components/DevCommand/SiteDev.tsx
var SiteDev = ({
  serverLocalUrl,
  serverExposedUrl,
  devMetadata
}) => {
  const { siteId } = useSite();
  const { tunnelRequired, devArmTag } = useDevCenterUpdater({
    serverLocalUrl,
    serverExposedUrl,
    devMetadata
  });
  return /* @__PURE__ */ import_react7.default.createElement(import_react7.default.Fragment, null, /* @__PURE__ */ import_react7.default.createElement(Text, null, "Development environment ready"), /* @__PURE__ */ import_react7.default.createElement(Box_default, { rowGap: 1, marginY: 1, flexDirection: "column" }, /* @__PURE__ */ import_react7.default.createElement(TunnelingConsent, { required: tunnelRequired }, /* @__PURE__ */ import_react7.default.createElement(Text, null, "Open the preview on:"), /* @__PURE__ */ import_react7.default.createElement(Box_default, { paddingLeft: 2, flexDirection: "column" }, /* @__PURE__ */ import_react7.default.createElement(PlatformLink, { url: serverLocalUrl, label: "Site" }), devArmTag ? /* @__PURE__ */ import_react7.default.createElement(
    PlatformLink,
    {
      url: `https://manage.wix.com/dashboard/${siteId}?apps-override=${devArmTag}`,
      label: "Dashboard"
    }
  ) : /* @__PURE__ */ import_react7.default.createElement(Spinner, { text: "Preparing your dashboard..." })))));
};

// ../cli-astro-commands/src/components/DevCommand/SiteDevFlow.tsx
var SiteDevFlow = ({
  devMetadataStatus,
  serverExposedUrl
}) => {
  const { siteId } = useSite();
  return /* @__PURE__ */ import_react8.default.createElement(SiteAuthProvider, { siteId }, (0, import_variant4.match)(devMetadataStatus, {
    NotRequested: () => /* @__PURE__ */ import_react8.default.createElement(Spinner, { text: "Starting dev environment..." }),
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react8.default.createElement(Spinner, { text: "Checking dev environment settings..." }),
    Success: ({ result: { devMetadata, serverLocalUrl } }) => /* @__PURE__ */ import_react8.default.createElement(DevScreenBorder, null, /* @__PURE__ */ import_react8.default.createElement(
      SiteDev,
      {
        devMetadata,
        serverLocalUrl,
        serverExposedUrl
      }
    ))
  }));
};

// ../cli-astro-commands/src/components/DevCommand/AppDevFlow.tsx
init_esm_shims();
var import_react14 = __toESM(require_react(), 1);
var import_variant7 = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/DevCommand/app-flow/AppDev.tsx
init_esm_shims();
var import_react13 = __toESM(require_react(), 1);
var import_variant6 = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/DevCommand/app-flow/screen-state.ts
init_esm_shims();
var import_variant5 = __toESM(require_lib(), 1);
var ScreenState = (0, import_variant5.variant)({
  Home: {},
  SiteSelector: {}
});

// ../cli-astro-commands/src/components/DevCommand/app-flow/SiteSelectorScreen.tsx
init_esm_shims();
var import_react10 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/CloseScreen/index.tsx
init_esm_shims();

// ../cli-astro-commands/src/components/CloseScreen/CloseScreen.tsx
init_esm_shims();
var import_react9 = __toESM(require_react(), 1);
var CloseScreen = ({ onClose }) => {
  const bi = useWixCliAppBi();
  use_input_default((_, key) => {
    if (key.escape) {
      bi.cliAction({
        question: "Press to close",
        questionKey: "general.close_screen",
        action: "close",
        key: "esc"
      });
      onClose();
    }
  });
  return /* @__PURE__ */ import_react9.default.createElement(Text, { skin: "secondary" }, "Press", /* @__PURE__ */ import_react9.default.createElement(Key, { value: "esc", skin: "secondary" }), "to close");
};

// ../cli-astro-commands/src/components/DevCommand/app-flow/SiteSelectorScreen.tsx
var SiteSelectorScreen = ({ onScreenChange }) => {
  const { model } = useProjectModel();
  const biLogger = useBiLogger();
  const { siteId, setSite } = useSite();
  const { reportError } = useErrorReporter();
  const [error, setError] = (0, import_react10.useState)(null);
  const handleError = (0, import_react10.useCallback)(
    (error2) => {
      reportError(error2);
      setError(error2);
    },
    [reportError, setError]
  );
  const handleSubmit = (0, import_react10.useCallback)(
    async (site) => {
      try {
        await setSite(site.id);
        onScreenChange(ScreenState.Home());
      } catch (e) {
        handleError(e);
      }
    },
    [setSite, onScreenChange, handleError]
  );
  return /* @__PURE__ */ import_react10.default.createElement(Box_default, { flexDirection: "column", gap: 1 }, /* @__PURE__ */ import_react10.default.createElement(Box_default, { gap: 1 }, /* @__PURE__ */ import_react10.default.createElement(PointerIcon, null), /* @__PURE__ */ import_react10.default.createElement(Text, null, "Current Development Site: ", siteId)), /* @__PURE__ */ import_react10.default.createElement(Box_default, { flexDirection: "column" }, /* @__PURE__ */ import_react10.default.createElement(
    SiteSelector,
    {
      appId: model.config.appId,
      biLogger,
      onSubmit: handleSubmit,
      onError: handleError
    }
  ), error ? /* @__PURE__ */ import_react10.default.createElement(
    ErrorViewer,
    {
      error,
      systemErrorOverride: /* @__PURE__ */ import_react10.default.createElement(Alert, { type: "error" }, "Failed to change Development Site!")
    }
  ) : null), /* @__PURE__ */ import_react10.default.createElement(CloseScreen, { onClose: () => onScreenChange(ScreenState.Home()) }));
};

// ../cli-astro-commands/src/components/DevCommand/app-flow/HomeScreen.tsx
init_esm_shims();
var import_react12 = __toESM(require_react(), 1);
import process from "node:process";

// ../cli-astro-commands/src/components/DevCommand/app-flow/SwitchSite.tsx
init_esm_shims();
var import_react11 = __toESM(require_react(), 1);
var SwitchSite = ({ onScreenChange }) => {
  const bi = useWixCliAppBi();
  use_input_default(async (input) => {
    if (input === "c") {
      bi.cliAction({
        question: "Switch Dev Site",
        questionKey: "dev_command.dev_site.change",
        action: "change_site",
        key: input
      });
      onScreenChange(ScreenState.SiteSelector());
    }
  });
  return /* @__PURE__ */ import_react11.default.createElement(Text, { skin: "secondary" }, "(Press ", /* @__PURE__ */ import_react11.default.createElement(Key, { value: "C", skin: "secondary" }), " to switch)");
};

// ../cli-astro-commands/src/components/DevCommand/app-flow/HomeScreen.tsx
var HomeScreen = ({
  onScreenChange,
  devArmTag
}) => {
  const { siteId } = useSite();
  return /* @__PURE__ */ import_react12.default.createElement(import_react12.default.Fragment, null, /* @__PURE__ */ import_react12.default.createElement(Box_default, { gap: 1 }, /* @__PURE__ */ import_react12.default.createElement(Text, null, "Current Dev Site:"), /* @__PURE__ */ import_react12.default.createElement(Text, { bold: true, skin: "info" }, siteId), process.stdin.isTTY && /* @__PURE__ */ import_react12.default.createElement(SwitchSite, { onScreenChange })), /* @__PURE__ */ import_react12.default.createElement(Box_default, { paddingTop: 1, paddingBottom: 1, flexDirection: "column" }, /* @__PURE__ */ import_react12.default.createElement(Text, null, "Open the preview on:"), /* @__PURE__ */ import_react12.default.createElement(Box_default, { paddingLeft: 2, flexDirection: "column" }, devArmTag ? /* @__PURE__ */ import_react12.default.createElement(import_react12.default.Fragment, null, /* @__PURE__ */ import_react12.default.createElement(
    RedirectLink,
    {
      platform: "site",
      armTag: devArmTag,
      tenantId: siteId
    }
  ), /* @__PURE__ */ import_react12.default.createElement(
    RedirectLink,
    {
      platform: "editor",
      armTag: devArmTag,
      tenantId: siteId
    }
  ), /* @__PURE__ */ import_react12.default.createElement(
    RedirectLink,
    {
      platform: "dashboard",
      armTag: devArmTag,
      tenantId: siteId
    }
  )) : /* @__PURE__ */ import_react12.default.createElement(Spinner, { text: "Preparing your dashboard..." }))));
};

// ../cli-astro-commands/src/components/DevCommand/app-flow/AppDev.tsx
var AppDev = ({
  serverLocalUrl,
  serverExposedUrl,
  devMetadata
}) => {
  const [screen, setScreen] = (0, import_react13.useState)(ScreenState.Home());
  const { tunnelRequired, devArmTag } = useDevCenterUpdater({
    serverLocalUrl,
    serverExposedUrl,
    devMetadata
  });
  return /* @__PURE__ */ import_react13.default.createElement(TunnelingConsent, { required: tunnelRequired }, (0, import_variant6.match)(screen, {
    Home: () => /* @__PURE__ */ import_react13.default.createElement(HomeScreen, { onScreenChange: setScreen, devArmTag }),
    SiteSelector: () => /* @__PURE__ */ import_react13.default.createElement(SiteSelectorScreen, { onScreenChange: setScreen })
  }));
};

// ../cli-astro-commands/src/components/DevCommand/AppDevFlow.tsx
var AppDevFlow = ({
  devMetadataStatus,
  serverExposedUrl
}) => {
  return (0, import_variant7.match)(devMetadataStatus, {
    NotRequested: () => /* @__PURE__ */ import_react14.default.createElement(Spinner, { text: "Starting dev environment..." }),
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react14.default.createElement(Spinner, { text: "Checking dev environment settings..." }),
    Success: ({ result: { devMetadata, serverLocalUrl } }) => {
      return /* @__PURE__ */ import_react14.default.createElement(DevScreenBorder, null, /* @__PURE__ */ import_react14.default.createElement(
        AppDev,
        {
          serverLocalUrl,
          serverExposedUrl,
          devMetadata
        }
      ));
    }
  });
};

// ../cli-astro-commands/src/components/DevCommand/DevCommand.tsx
var DevCommand = ({
  port,
  allowedHosts,
  serverExposedUrl
}) => {
  const {
    model: { config, projectFolder }
  } = useProjectModel();
  const exit = useExit();
  const logger = (0, import_react15.useMemo)(() => createAstroLogger(false), []);
  const astroProcess = (0, import_react15.useRef)();
  const fetchDevMetadataAction = useFetchDevMetadata();
  useAsync(async () => {
    const repoType = await getRepoType(projectFolder);
    const packageManager = await createPackageManager(repoType);
    const astro = runAstroDev({
      packageManager,
      projectFolder,
      port,
      allowedHosts
    });
    astroProcess.current = astro;
    astro.stdout?.once(EVENT_SERVER_ADDRESS, (address) => {
      void fetchDevMetadataAction.execute(address);
    });
    astro.stdout?.on(EVENT_LINE, (line) => logger.log(line));
    astro.stderr?.on(EVENT_LINE, (line) => logger.log(line));
    void astro.on("exit", () => {
      astroProcess.current?.stdout?.destroy();
      astroProcess.current?.stderr?.destroy();
      astroProcess.current?.stdin?.end();
      astroProcess.current = void 0;
      exit(
        new CliError({
          cause: void 0,
          code: CliErrorCode.DevCommandFailed()
        })
      );
    });
  }, []);
  (0, import_react15.useEffect)(() => {
    return () => {
      if (astroProcess.current?.pid) {
        try {
          astroProcess.current.kill("SIGTERM");
        } catch {
        }
      }
    };
  }, []);
  return /* @__PURE__ */ import_react15.default.createElement(SiteProvider, null, config.projectType === "Site" ? /* @__PURE__ */ import_react15.default.createElement(
    SiteDevFlow,
    {
      devMetadataStatus: fetchDevMetadataAction.status,
      serverExposedUrl
    }
  ) : /* @__PURE__ */ import_react15.default.createElement(
    AppDevFlow,
    {
      devMetadataStatus: fetchDevMetadataAction.status,
      serverExposedUrl
    }
  ));
};
export {
  DevCommand
};
