// @wix/cli/build/InstallCommand-573BPGDW.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  createDependenciesService
} from "./chunk-ZEI53JZK.js";
import "./chunk-R5JDVBC7.js";
import {
  useVeloClient
} from "./chunk-DT3R4WAE.js";
import "./chunk-BXOQU5DN.js";
import "./chunk-Y7QTUVGN.js";
import "./chunk-B5TSS2PZ.js";
import "./chunk-MVBGFP6D.js";
import {
  useProjectModel
} from "./chunk-NPAOMGSP.js";
import "./chunk-AT5WRPKM.js";
import "./chunk-2R5BII2J.js";
import "./chunk-W54PG6O5.js";
import "./chunk-YLWTNITT.js";
import "./chunk-RHVTQ2DQ.js";
import "./chunk-RDT7NRVZ.js";
import "./chunk-3Q7XZST5.js";
import "./chunk-WMCHVSMU.js";
import {
  Trans
} from "./chunk-6WEX76B7.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Badge,
  Spinner,
  Text,
  useAsync
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKJGN2XT.js";
import "./chunk-C5PTSUGF.js";
import "./chunk-WDXANXXO.js";
import "./chunk-2OYCKIQP.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site/src/components/InstallCommand/InstallCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var InstallCommand = ({ packageName }) => {
  const { model } = useProjectModel();
  const veloClient = useVeloClient();
  const { status } = useAsync(async () => {
    const { projectFolder } = model;
    const service = await createDependenciesService(projectFolder, veloClient);
    await service.install(packageName);
  }, []);
  return (0, import_variant.match)(status, {
    Loading: () => /* @__PURE__ */ import_react.default.createElement(
      Spinner,
      {
        text: packageName ? /* @__PURE__ */ import_react.default.createElement(
          Trans,
          {
            i18nKey: "install_command.package.loading",
            values: { packageName },
            components: [/* @__PURE__ */ import_react.default.createElement(Text, { bold: true })]
          }
        ) : /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "install_command.all_packages.loading" })
      }
    ),
    Error: () => null,
    Success: () => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(Badge, { skin: "success" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "general.success" })), " ", packageName ? /* @__PURE__ */ import_react.default.createElement(
      Trans,
      {
        i18nKey: "install_command.package.success",
        values: { packageName },
        components: [/* @__PURE__ */ import_react.default.createElement(Text, { bold: true })]
      }
    ) : /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "install_command.all_packages.success" }))
  });
};
export {
  InstallCommand
};
