// @wix/cli/build/LoginCommand-Y2M65O6Y.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import "./chunk-FGGA4ZJX.js";
import {
  addUserFields
} from "./chunk-ZWLXMEDA.js";
import {
  ApiKeyLogin,
  LoginWithI18n,
  useBiLogger,
  useErrorReporter
} from "./chunk-3Q7XZST5.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-6WEX76B7.js";
import "./chunk-KJ37XZQA.js";
import {
  ConfirmInput,
  Text,
  useExit
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKJGN2XT.js";
import "./chunk-C5PTSUGF.js";
import "./chunk-WDXANXXO.js";
import "./chunk-2OYCKIQP.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// src/components/LoginCommand/index.ts
init_esm_shims();

// src/components/LoginCommand/LoginCommand.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);

// src/components/LoginCommand/AlreadyLoggedIn.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var AlreadyLoggedIn = ({ userInfo, onSubmit }) => {
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(Text, null, "Wix CLI is currently logged in to account", " ", /* @__PURE__ */ import_react.default.createElement(Text, { skin: "info" }, userInfo.email)), /* @__PURE__ */ import_react.default.createElement(
    ConfirmInput,
    {
      label: "Do you want to log in to a different account?",
      initialValue: true,
      onSubmit
    }
  ));
};

// src/components/LoginCommand/LoginCommand.tsx
var LoginState = (0, import_variant.variant)({
  PendingUserConfirmation: {},
  InProgress: {},
  Success: (0, import_variant.fields)()
});
var LoginCommand = ({
  apiKeyToken,
  userInfo: initialUserInfo
}) => {
  const exit = useExit();
  const errorReporter = useErrorReporter();
  const biLogger = useBiLogger();
  const [loginState, setLoginState] = (0, import_react2.useState)(
    initialUserInfo == null ? LoginState.InProgress() : LoginState.PendingUserConfirmation()
  );
  const handleLogin = (0, import_react2.useCallback)(
    ({ userInfo }) => {
      errorReporter.setUser({ id: userInfo.userId });
      addUserFields(biLogger, userInfo.userId);
      setLoginState(LoginState.Success({ userInfo }));
    },
    [setLoginState, errorReporter, biLogger]
  );
  return /* @__PURE__ */ import_react2.default.createElement(import_react2.default.Fragment, null, initialUserInfo && /* @__PURE__ */ import_react2.default.createElement(
    AlreadyLoggedIn,
    {
      userInfo: initialUserInfo,
      onSubmit: (relogin) => relogin && setLoginState(LoginState.InProgress())
    }
  ), (0, import_variant.match)(loginState, {
    PendingUserConfirmation: () => null,
    InProgress: () => apiKeyToken ? /* @__PURE__ */ import_react2.default.createElement(
      ApiKeyLogin,
      {
        token: apiKeyToken,
        onLogin: handleLogin,
        onError: exit
      }
    ) : /* @__PURE__ */ import_react2.default.createElement(LoginWithI18n, { onLogin: handleLogin, onError: exit }),
    Success: ({ userInfo }) => /* @__PURE__ */ import_react2.default.createElement(Text, null, "Wix CLI is logged in to account", " ", /* @__PURE__ */ import_react2.default.createElement(Text, { skin: "info" }, userInfo.email))
  }));
};
export {
  LoginCommand
};
