// @wix/cli/build/PromoteCommand-QAOZIIXY.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  PreviewCommand
} from "./chunk-LCGBY36X.js";
import {
  useARMClient
} from "./chunk-CZ76ZPHP.js";
import "./chunk-42GS4QEG.js";
import "./chunk-SC2W3WT3.js";
import "./chunk-AT5WRPKM.js";
import "./chunk-2R5BII2J.js";
import "./chunk-W54PG6O5.js";
import "./chunk-YLWTNITT.js";
import "./chunk-SKN7SRA5.js";
import "./chunk-MEY2AT7Z.js";
import {
  useProjectModel
} from "./chunk-OPNNOJNG.js";
import "./chunk-RHVTQ2DQ.js";
import "./chunk-PIXNEJ52.js";
import "./chunk-HWVLWVTF.js";
import "./chunk-RDT7NRVZ.js";
import "./chunk-3Q7XZST5.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-6WEX76B7.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Box_default,
  Spinner,
  Text,
  useAsyncCallback
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKJGN2XT.js";
import "./chunk-C5PTSUGF.js";
import "./chunk-WDXANXXO.js";
import "./chunk-2OYCKIQP.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/PromoteCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/PromoteCommand/PromoteCommand.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/PromoteCommand/Steps/CreatePreview.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var CreatePreview = (props) => {
  const { baseUrl, label, onCreated } = props;
  return /* @__PURE__ */ import_react.default.createElement(Box_default, { flexDirection: "column", gap: 1 }, /* @__PURE__ */ import_react.default.createElement(Text, null, "Before promoting, let's create a preview of your app"), /* @__PURE__ */ import_react.default.createElement(Box_default, { flexDirection: "column" }, /* @__PURE__ */ import_react.default.createElement(
    PreviewCommand,
    {
      baseUrl,
      label,
      onPreviewCreated: onCreated,
      environment: "preview"
    }
  )));
};

// ../cli-astro-commands/src/components/PromoteCommand/PromoteCommand.tsx
var PromoteCommand = ({ baseUrl, label }) => {
  const {
    model: {
      config: { appId }
    }
  } = useProjectModel();
  const arm = useARMClient();
  const promoteAction = useAsyncCallback(async (_, tag) => {
    await arm.createDraftVersion(appId, tag);
  });
  function onPreviewCreated({ previewId }) {
    return promoteAction.execute(previewId);
  }
  return /* @__PURE__ */ import_react2.default.createElement(import_react2.default.Fragment, null, /* @__PURE__ */ import_react2.default.createElement(
    CreatePreview,
    {
      baseUrl,
      label,
      onCreated: ({ previewId }) => onPreviewCreated({ previewId })
    }
  ), (0, import_variant.match)(promoteAction.status, {
    Loading: () => /* @__PURE__ */ import_react2.default.createElement(Spinner, { text: `Promoting app draft...` }),
    Success: () => /* @__PURE__ */ import_react2.default.createElement(Alert, { type: "success" }, "App draft promoted successfully"),
    Error: () => null,
    NotRequested: () => null
  }));
};
export {
  PromoteCommand
};
