// @wix/cli/build/PublishCommand-NCLOO6KA.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useCreatePreview
} from "./chunk-OEVPUQPU.js";
import {
  DeploymentStatus,
  useDocumentManagementClient
} from "./chunk-Q4OSIGEB.js";
import "./chunk-NQAF6TXV.js";
import "./chunk-CH52GF36.js";
import "./chunk-T5LNGJQV.js";
import "./chunk-R5JDVBC7.js";
import "./chunk-DT3R4WAE.js";
import "./chunk-BXOQU5DN.js";
import "./chunk-Y7QTUVGN.js";
import "./chunk-B5TSS2PZ.js";
import "./chunk-MVBGFP6D.js";
import "./chunk-NPAOMGSP.js";
import "./chunk-RHVTQ2DQ.js";
import "./chunk-RDT7NRVZ.js";
import {
  ErrorViewer
} from "./chunk-3Q7XZST5.js";
import "./chunk-WMCHVSMU.js";
import {
  Trans,
  useTranslation
} from "./chunk-6WEX76B7.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Badge,
  Box_default,
  ConfirmInput,
  Link,
  Spinner,
  useAsync,
  useAsyncCallback
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKJGN2XT.js";
import "./chunk-C5PTSUGF.js";
import "./chunk-WDXANXXO.js";
import "./chunk-2OYCKIQP.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site/src/components/PublishCommand/PublishCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var PublishCommand = ({ approvePreview, force }) => {
  const { t } = useTranslation();
  const createPreview = useCreatePreview();
  const { publishDeployment } = useDocumentManagementClient();
  const publishDeploymentAction = useAsyncCallback(
    async (_, deploymentId) => publishDeployment(deploymentId)
  );
  const { status } = useAsync(async () => {
    const preview = await createPreview();
    const preApprovePreview = force || ((0, import_variant.isType)(preview.deploymentStatus, DeploymentStatus.Success) ? approvePreview : false);
    if (preApprovePreview) {
      void publishDeploymentAction.execute(preview.deploymentId);
    }
    return {
      ...preview,
      preApprovePreview
    };
  }, []);
  return (0, import_variant.match)(status, {
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "publish_command.preview.loading" }) }),
    Success: ({ result }) => {
      return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, (0, import_variant.match)(result.deploymentStatus, {
        Success: () => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(
          Trans,
          {
            i18nKey: "publish_command.preview.success",
            components: [/* @__PURE__ */ import_react.default.createElement(Link, { url: result.previewUrl })]
          }
        )),
        Error: ({ deploymentError }) => /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(ErrorViewer, { error: deploymentError }), /* @__PURE__ */ import_react.default.createElement(Box_default, { marginTop: 1 }, /* @__PURE__ */ import_react.default.createElement(Alert, { type: "warning" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "publish_command.preview.warning" }))))
      }), !result.preApprovePreview && /* @__PURE__ */ import_react.default.createElement(Box_default, { marginTop: 1 }, /* @__PURE__ */ import_react.default.createElement(
        ConfirmInput,
        {
          label: t("publish_command.confirm_publish"),
          onSubmit: (value) => {
            if (value) {
              void publishDeploymentAction.execute(result.deploymentId);
            }
          }
        }
      )), (0, import_variant.match)(publishDeploymentAction.status, {
        NotRequested: () => null,
        Error: () => null,
        Loading: () => /* @__PURE__ */ import_react.default.createElement(Box_default, { marginTop: 1 }, /* @__PURE__ */ import_react.default.createElement(
          Spinner,
          {
            text: /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "publish_command.publish.loading" })
          }
        )),
        Success: () => /* @__PURE__ */ import_react.default.createElement(Box_default, { marginTop: 1 }, /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(Badge, { skin: "success" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "general.success" })), " ", /* @__PURE__ */ import_react.default.createElement(
          Trans,
          {
            i18nKey: "publish_command.publish.success",
            components: [/* @__PURE__ */ import_react.default.createElement(Link, { url: result.siteUrl })]
          }
        )))
      }));
    }
  });
};
export {
  PublishCommand
};
