// @wix/cli/build/ReleaseCommand-JHFUWMKH.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  ReleaseVersionType
} from "./chunk-2IJTJJEM.js";
import {
  PreviewCommand,
  assertBuildOutputExists
} from "./chunk-LCGBY36X.js";
import {
  useWixCliAppBi
} from "./chunk-CW6TLWH7.js";
import {
  useARMClient
} from "./chunk-CZ76ZPHP.js";
import "./chunk-42GS4QEG.js";
import "./chunk-SC2W3WT3.js";
import "./chunk-AT5WRPKM.js";
import "./chunk-2R5BII2J.js";
import "./chunk-W54PG6O5.js";
import "./chunk-YLWTNITT.js";
import "./chunk-SKN7SRA5.js";
import {
  require_isArray
} from "./chunk-MEY2AT7Z.js";
import {
  ComponentType,
  useProjectModel
} from "./chunk-OPNNOJNG.js";
import "./chunk-RHVTQ2DQ.js";
import "./chunk-PIXNEJ52.js";
import "./chunk-HWVLWVTF.js";
import "./chunk-RDT7NRVZ.js";
import {
  require_semver
} from "./chunk-3Q7XZST5.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-6WEX76B7.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Box_default,
  Cart,
  ConfirmInput,
  Link,
  Notification,
  SelectInput,
  SelectOptionAvailability,
  Spinner,
  Statistics,
  Text,
  TextInput,
  require_baseGetTag,
  require_isObjectLike,
  useAsync,
  useAsyncCallback
} from "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKJGN2XT.js";
import "./chunk-C5PTSUGF.js";
import "./chunk-WDXANXXO.js";
import "./chunk-2OYCKIQP.js";
import "./chunk-F43XHKVL.js";
import {
  CliError,
  CliErrorCode,
  require_lib
} from "./chunk-HNXJYLVA.js";
import {
  __commonJS,
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../../node_modules/lodash/isString.js
var require_isString = __commonJS({
  "../../node_modules/lodash/isString.js"(exports, module) {
    "use strict";
    init_esm_shims();
    var baseGetTag = require_baseGetTag();
    var isArray = require_isArray();
    var isObjectLike = require_isObjectLike();
    var stringTag = "[object String]";
    function isString2(value) {
      return typeof value == "string" || !isArray(value) && isObjectLike(value) && baseGetTag(value) == stringTag;
    }
    module.exports = isString2;
  }
});

// ../cli-astro-commands/src/components/ReleaseCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/ReleaseCommand/ReleaseCommand.tsx
init_esm_shims();
var import_react10 = __toESM(require_react(), 1);
var import_variant3 = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/ReleaseCommand/Release.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/ReleaseCommand/Steps/CreatePreview.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var CreatePreview = (props) => {
  const { baseUrl, label, onCreated } = props;
  return /* @__PURE__ */ import_react.default.createElement(Box_default, { flexDirection: "column", gap: 1 }, /* @__PURE__ */ import_react.default.createElement(Text, null, "Before creating a new version, let's create a preview of your app"), /* @__PURE__ */ import_react.default.createElement(Box_default, { flexDirection: "column" }, /* @__PURE__ */ import_react.default.createElement(
    PreviewCommand,
    {
      baseUrl,
      label,
      onPreviewCreated: onCreated,
      environment: "production"
    }
  )));
};

// ../cli-astro-commands/src/components/ReleaseCommand/Release.tsx
var Release = ({
  baseUrl,
  label,
  createMinorVersion,
  comment
}) => {
  const {
    model: {
      config: { appId }
    }
  } = useProjectModel();
  const { release } = useARMClient();
  const releaseAction = useAsyncCallback(async (_, tag) => {
    const baseUrl2 = await release(appId, tag, {
      createMinorVersion,
      comment
    });
    return {
      baseUrl: baseUrl2
    };
  });
  function onPreviewCreated({ previewId }) {
    return releaseAction.execute(previewId);
  }
  return /* @__PURE__ */ import_react2.default.createElement(import_react2.default.Fragment, null, /* @__PURE__ */ import_react2.default.createElement(
    CreatePreview,
    {
      baseUrl,
      label,
      onCreated: ({ previewId }) => onPreviewCreated({ previewId })
    }
  ), (0, import_variant.match)(releaseAction.status, {
    NotRequested: () => null,
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react2.default.createElement(Spinner, { text: /* @__PURE__ */ import_react2.default.createElement(Text, null, "Releasing...") }),
    Success: ({ result: { baseUrl: baseUrl2 } }) => /* @__PURE__ */ import_react2.default.createElement(Box_default, { flexDirection: "column", gap: 1 }, /* @__PURE__ */ import_react2.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react2.default.createElement(Text, { bold: true }, "Site published on ", baseUrl2)))
  }));
};

// ../cli-astro-commands/src/components/ReleaseCommand/app-flow/ReleaseApp.tsx
init_esm_shims();
var import_react9 = __toESM(require_react(), 1);
var import_isString = __toESM(require_isString(), 1);
var import_semver = __toESM(require_semver(), 1);
var import_variant2 = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/ReleaseCommand/app-flow/Warnings.tsx
init_esm_shims();
var import_react4 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/extensions/useExtensions.tsx
init_esm_shims();
var import_react3 = __toESM(require_react(), 1);
function useExtensions() {
  const { model } = useProjectModel();
  const getExtensions = (0, import_react3.useCallback)(
    (extensionType) => model.components.filter(
      (component) => component.compType === extensionType
    ),
    [model.components]
  );
  return {
    getExtensions
  };
}

// ../cli-astro-commands/src/components/ReleaseCommand/app-flow/Warnings.tsx
var Warnings = () => {
  const { getExtensions } = useExtensions();
  if (getExtensions(ComponentType.WEBHOOK).length === 0) {
    return null;
  }
  return /* @__PURE__ */ import_react4.default.createElement(Box_default, { flexDirection: "column", paddingBottom: 1 }, /* @__PURE__ */ import_react4.default.createElement(Notification, null, /* @__PURE__ */ import_react4.default.createElement(Alert, { type: "warning" }, "Your project includes backend events. Once you proceed, any changes to events will be immediately deployed to production and take effect automatically, even without a new version release.")));
};

// ../cli-astro-commands/src/components/ReleaseCommand/app-flow/ReleaseComment.tsx
init_esm_shims();
var import_react5 = __toESM(require_react(), 1);
var ReleaseComment = (props) => {
  const { onEntered } = props;
  const bi = useWixCliAppBi();
  const validate = (0, import_react5.useCallback)((value) => {
    if (value && value.length > 250) {
      return "The comment is limited to a maximum of 250 characters";
    }
    return true;
  }, []);
  const question = "Enter a comment describing any changes made to this version. This won't be shown to users (optional)";
  return /* @__PURE__ */ import_react5.default.createElement(
    TextInput,
    {
      label: question,
      onSubmit: (answer) => {
        bi.cliFlowStepAnswered({
          question,
          questionKey: "release_command.enter_release_comment",
          answer
        });
        onEntered(answer);
      },
      validate
    }
  );
};

// ../cli-astro-commands/src/components/ReleaseCommand/app-flow/ConfirmMinorVersion.tsx
init_esm_shims();
var import_react6 = __toESM(require_react(), 1);
var ConfirmMinorVersion = (props) => {
  const { onConfirmed } = props;
  const bi = useWixCliAppBi();
  const question = "Changes will be visible to your app users immediately and can\u2019t be undone. Do you want to continue?";
  return /* @__PURE__ */ import_react6.default.createElement(Box_default, { flexDirection: "column", paddingBottom: 1 }, /* @__PURE__ */ import_react6.default.createElement(
    ConfirmInput,
    {
      label: question,
      initialValue: true,
      onSubmit: (confirm) => {
        bi.cliFlowStepAnswered({
          question,
          questionKey: "release_command.confirm_version_creation",
          answer: String(confirm)
        });
        if (confirm) {
          onConfirmed();
        }
      }
    }
  ));
};

// ../cli-astro-commands/src/components/ReleaseCommand/app-flow/ChooseVersion.tsx
init_esm_shims();
var import_react7 = __toESM(require_react(), 1);
var ChooseVersion = (props) => {
  const { nextMinor, nextMajor, onChosen } = props;
  const bi = useWixCliAppBi();
  const options = [
    {
      title: nextMinor ? `Minor (${nextMinor})` : "Minor",
      description: "update doesn\u2019t require Wix user consent",
      value: ReleaseVersionType.MINOR,
      availability: nextMinor ? SelectOptionAvailability.Enabled() : SelectOptionAvailability.Disabled({
        reason: /* @__PURE__ */ import_react7.default.createElement(Text, null, "Recent changes you made require this release to be a major release. ", /* @__PURE__ */ import_react7.default.createElement(Link, { url: "https://wix.to/ZR80Y7p" }, "Learn more"))
      })
    },
    {
      title: `Major (${nextMajor})`,
      description: "update requires Wix user consent",
      value: ReleaseVersionType.MAJOR
    }
  ];
  const question = "What type of version would you like to create?";
  return /* @__PURE__ */ import_react7.default.createElement(
    SelectInput,
    {
      label: question,
      options,
      initialIndex: nextMinor ? 0 : 1,
      onSubmit: ({ value }) => {
        bi.cliFlowStepAnswered({
          question,
          questionKey: "release_command.version_type_question",
          answer: value
        });
        onChosen(value);
      }
    }
  );
};

// ../cli-astro-commands/src/components/ReleaseCommand/app-flow/NextSteps.tsx
init_esm_shims();
var import_react8 = __toESM(require_react(), 1);
var NextSteps = () => {
  const {
    model: {
      config: { appId }
    }
  } = useProjectModel();
  const homeUrl = `https://dev.wix.com/apps/${appId}/home?referralInfo=wix-cli`;
  const statisticsUrl = `https://dev.wix.com/apps/${appId}/statistics?referralInfo=wix-cli`;
  return /* @__PURE__ */ import_react8.default.createElement(Box_default, { flexDirection: "column", gap: 1 }, /* @__PURE__ */ import_react8.default.createElement(Text, { bold: true }, "Next Steps:"), /* @__PURE__ */ import_react8.default.createElement(Text, null, /* @__PURE__ */ import_react8.default.createElement(Statistics, null), " View your app reviews and statistics in the", " ", /* @__PURE__ */ import_react8.default.createElement(Link, { url: statisticsUrl }, "app dashboard")), /* @__PURE__ */ import_react8.default.createElement(Text, null, /* @__PURE__ */ import_react8.default.createElement(Cart, null), " Choose your distribution method from the", " ", /* @__PURE__ */ import_react8.default.createElement(Link, { url: homeUrl }, "app dashboard")));
};

// ../cli-astro-commands/src/components/ReleaseCommand/app-flow/ReleaseApp.tsx
var ReleaseApp = ({
  baseUrl,
  label,
  preEnteredComment,
  preSelectedVersionType
}) => {
  const {
    model: {
      config: { appId }
    }
  } = useProjectModel();
  const { createDraftVersion, releaseDraftVersion } = useARMClient();
  const [tag, setTag] = (0, import_react9.useState)();
  const [comment, setComment] = (0, import_react9.useState)();
  const [versionType, setVersionType] = (0, import_react9.useState)();
  const [userShouldSelectVersion, setUserShouldSelectVersion] = (0, import_react9.useState)(false);
  const createDraftAction = useAsyncCallback(async (_, tag2) => {
    const { nextMajor, nextMinor } = await createDraftVersion(appId, tag2);
    const appWasNeverReleased = nextMajor === "1.0.0" && !nextMinor;
    const parsedNextMajor = (0, import_semver.parse)(nextMajor);
    if (parsedNextMajor == null) {
      throw new CliError({
        code: CliErrorCode.FailedToParseNextVersion({ version: nextMajor }),
        cause: void 0
      });
    }
    return {
      nextMinor,
      nextMajor,
      appWasNeverReleased,
      latestMajor: parsedNextMajor.major - 1
    };
  });
  const releaseDraftAction = useAsyncCallback(
    async (_, versionType2, comment2, latestMajor) => {
      const result = await releaseDraftVersion(appId, {
        createNewMajor: versionType2 === ReleaseVersionType.MAJOR,
        notes: comment2,
        latestMajor
      });
      return result;
    }
  );
  async function onReleaseCommentEntered({
    tag: tag2,
    comment: comment2
  }) {
    setComment(comment2);
    const result = await createDraftAction.execute(tag2).catch(() => null);
    if (!result) {
      return;
    }
    const { appWasNeverReleased, nextMinor, latestMajor } = result;
    const chosenVersion = appWasNeverReleased ? ReleaseVersionType.MAJOR : preSelectedVersionType;
    if (preSelectedVersionType === ReleaseVersionType.MINOR && (appWasNeverReleased || !nextMinor)) {
      throw new CliError({
        cause: null,
        code: CliErrorCode.CannotReleaseMinorInNoninteractive()
      });
    }
    if (chosenVersion) {
      void releaseDraftAction.execute(chosenVersion, comment2, latestMajor);
    } else {
      setUserShouldSelectVersion(true);
    }
  }
  return /* @__PURE__ */ import_react9.default.createElement(import_react9.default.Fragment, null, /* @__PURE__ */ import_react9.default.createElement(
    CreatePreview,
    {
      baseUrl,
      label,
      onCreated: ({ previewId }) => {
        setTag(previewId);
        if ((0, import_isString.default)(preEnteredComment)) {
          void onReleaseCommentEntered({
            tag: previewId,
            comment: preEnteredComment
          });
        }
      }
    }
  ), tag && /* @__PURE__ */ import_react9.default.createElement(Warnings, null), tag && !(0, import_isString.default)(preEnteredComment) && /* @__PURE__ */ import_react9.default.createElement(
    ReleaseComment,
    {
      onEntered: (comment2) => onReleaseCommentEntered({ tag, comment: comment2 })
    }
  ), (0, import_isString.default)(comment) && (0, import_variant2.match)(createDraftAction.status, {
    NotRequested: () => null,
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react9.default.createElement(Spinner, { text: "Updating draft application version" }),
    Success: ({ result: { nextMajor, nextMinor, latestMajor } }) => /* @__PURE__ */ import_react9.default.createElement(import_react9.default.Fragment, null, userShouldSelectVersion && /* @__PURE__ */ import_react9.default.createElement(
      ChooseVersion,
      {
        nextMinor,
        nextMajor,
        onChosen: (selectedVersionType) => {
          setVersionType(selectedVersionType);
          if (selectedVersionType === ReleaseVersionType.MAJOR) {
            void releaseDraftAction.execute(
              selectedVersionType,
              comment,
              latestMajor
            );
          }
        }
      }
    ), versionType === ReleaseVersionType.MINOR && /* @__PURE__ */ import_react9.default.createElement(
      ConfirmMinorVersion,
      {
        onConfirmed: () => releaseDraftAction.execute(
          versionType,
          comment,
          latestMajor
        )
      }
    ))
  }), (0, import_variant2.match)(releaseDraftAction.status, {
    NotRequested: () => null,
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react9.default.createElement(Spinner, { text: "Releasing..." }),
    Success: ({ result: { releasedVersion } }) => /* @__PURE__ */ import_react9.default.createElement(Box_default, { flexDirection: "column", gap: 1 }, /* @__PURE__ */ import_react9.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react9.default.createElement(Text, { bold: true }, "Version ", releasedVersion, " was created")), /* @__PURE__ */ import_react9.default.createElement(NextSteps, null))
  }));
};

// ../cli-astro-commands/src/components/ReleaseCommand/ReleaseCommand.tsx
var ReleaseCommand = (props) => {
  const {
    model: { projectFolder, config }
  } = useProjectModel();
  const { status } = useAsync(async () => {
    await assertBuildOutputExists(projectFolder);
  }, []);
  return (0, import_variant3.match)(status, {
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react10.default.createElement(Spinner, { text: /* @__PURE__ */ import_react10.default.createElement(Text, null, "Loading application details...") }),
    Success: () => {
      return config.projectType === "Site" ? /* @__PURE__ */ import_react10.default.createElement(
        Release,
        {
          ...props,
          createMinorVersion: props.versionType === ReleaseVersionType.MINOR
        }
      ) : /* @__PURE__ */ import_react10.default.createElement(
        ReleaseApp,
        {
          baseUrl: props.baseUrl,
          label: props.label,
          preEnteredComment: props.comment,
          preSelectedVersionType: props.versionType
        }
      );
    }
  });
};
export {
  ReleaseCommand
};
