// @wix/cli/build/src-SDMILJU2.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  docsExtensionTypeSchema
} from "./chunk-FRO7EYZX.js";
import {
  ReleaseVersionType
} from "./chunk-2IJTJJEM.js";
import {
  renderCommand,
  runCommand
} from "./chunk-FUJVYDK4.js";
import "./chunk-OPNNOJNG.js";
import "./chunk-RHVTQ2DQ.js";
import {
  NO_TTY_JSON_OUTPUT_OPTION
} from "./chunk-AWPJTA2S.js";
import {
  InvalidArgumentError,
  Option,
  createCommand
} from "./chunk-J2EPR7DO.js";
import "./chunk-RDT7NRVZ.js";
import "./chunk-ZWLXMEDA.js";
import "./chunk-3Q7XZST5.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-6WEX76B7.js";
import "./chunk-KJ37XZQA.js";
import "./chunk-3SLQR6EP.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKJGN2XT.js";
import "./chunk-C5PTSUGF.js";
import "./chunk-WDXANXXO.js";
import "./chunk-2OYCKIQP.js";
import {
  external_exports
} from "./chunk-F43XHKVL.js";
import "./chunk-HNXJYLVA.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/index.ts
init_esm_shims();

// ../cli-astro-commands/src/commands/root-command.ts
init_esm_shims();

// ../cli-astro-commands/src/commands/build.ts
init_esm_shims();
var buildCommand = (services, cliVersion) => createCommand("build").description(
  "Invoke the astro build process, forwarding all command-line arguments."
).allowExcessArguments(true).allowUnknownOption(true).passThroughOptions(true).action(async (_, command) => {
  const { execa } = await import("./execa-MR2E3266.js");
  const { getRepoType, createPackageManager } = await import("./src-GKTAVUM4.js");
  const { runCommand: runCommand2 } = await import("./render-command-3NAFANF3.js");
  await runCommand2(command, services, cliVersion, async (model) => {
    const repoType = await getRepoType(model.projectFolder);
    const packageManager = await createPackageManager(repoType);
    const { file, args } = packageManager.getRunBinaryCmd(
      ["astro", "build"].concat(command.args)
    );
    await execa(file, args, {
      cwd: model.projectFolder,
      stdio: "inherit"
    });
  });
});

// ../cli-astro-commands/src/commands/env.ts
init_esm_shims();

// ../cli-astro-commands/src/commands/env-pull.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var envPullCommand = (services, cliVersion) => {
  return createCommand("pull").description(
    "Pulls environment variables and merge them into .env.local file"
  ).addOption(
    new Option(NO_TTY_JSON_OUTPUT_OPTION, "Output in JSON format").hideHelp()
  ).action(async (opts, command) => {
    if (opts.json) {
      return runCommand(command, services, cliVersion, async (model) => {
        const { nonInteractiveEnvPull } = await import("./env-pull-2QGFYUJP.js");
        return nonInteractiveEnvPull(services, model);
      });
    }
    const { EnvPullCommand } = await import("./EnvPullCommand-EY6JTTPE.js");
    await renderCommand(command, services, cliVersion, /* @__PURE__ */ import_react.default.createElement(EnvPullCommand, null));
  });
};

// ../cli-astro-commands/src/commands/env-set.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);
var envSetCommand = (services, cliVersion) => {
  return createCommand("set").description("Sets environment variable").requiredOption("--key <key>", "Name of the variable").requiredOption("--value <value>", "Value of the variable").action(async (opts, command) => {
    const { EnvSetCommand } = await import("./EnvSetCommand-IR4KMWQA.js");
    const variable = { key: opts.key, value: opts.value };
    await renderCommand(
      command,
      services,
      cliVersion,
      /* @__PURE__ */ import_react2.default.createElement(EnvSetCommand, { variable })
    );
  });
};

// ../cli-astro-commands/src/commands/env-remove.tsx
init_esm_shims();
var import_react3 = __toESM(require_react(), 1);
var envRemoveCommand = (services, cliVersion) => {
  return createCommand("remove").description("Removes an environment variable").requiredOption("--key <key>", "Name of the variable").action(async (opts, command) => {
    const { EnvRemoveCommand } = await import("./EnvRemoveCommand-FGDAH4BV.js");
    await renderCommand(
      command,
      services,
      cliVersion,
      /* @__PURE__ */ import_react3.default.createElement(EnvRemoveCommand, { variableKey: opts.key })
    );
  });
};

// ../cli-astro-commands/src/commands/env.ts
var envCommand = (services, cliVersion) => {
  return createCommand("env").description("Work with environment variables").addCommand(envPullCommand(services, cliVersion)).addCommand(envSetCommand(services, cliVersion)).addCommand(envRemoveCommand(services, cliVersion));
};

// ../cli-astro-commands/src/commands/preview.tsx
init_esm_shims();
var import_react4 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/commands/validations.ts
init_esm_shims();
function validateBase(value) {
  const parsed = external_exports.string().url().safeParse(value);
  if (parsed.success) {
    return parsed.data;
  }
  throw new InvalidArgumentError("Value must be a valid URL");
}
var LABEL_MAX_LENGTH = 200;
function validateLabel(value) {
  const parsed = external_exports.string().max(LABEL_MAX_LENGTH).safeParse(value);
  if (parsed.success) {
    return parsed.data;
  }
  throw new InvalidArgumentError(
    `Value must be up to ${LABEL_MAX_LENGTH} characters`
  );
}
function validateReleaseComment(value) {
  const parsed = external_exports.string().max(250).safeParse(value);
  if (parsed.success) {
    return parsed.data;
  }
  throw new InvalidArgumentError(
    "The comment is limited to a maximum of 250 characters"
  );
}
function validateDocsExtensionType(value) {
  const parsed = docsExtensionTypeSchema.safeParse(value);
  if (parsed.success) {
    return parsed.data;
  }
  throw new InvalidArgumentError("Value must be a valid extension type");
}

// ../cli-astro-commands/src/commands/preview.tsx
var previewCommand = (services, cliVersion) => {
  return createCommand("preview").addOption(
    new Option(
      "--base-url <url>",
      "The base url for when static files have been uploaded to an external cdn"
    ).argParser(validateBase).hideHelp()
  ).addOption(
    new Option("-l, --label <label>", "A custom label for this preview").argParser(validateLabel).hideHelp()
  ).addOption(new Option("--site <site>", "(deprecated)").hideHelp()).addOption(
    new Option(NO_TTY_JSON_OUTPUT_OPTION, "Output in JSON format").hideHelp()
  ).action(async (options, command) => {
    const { baseUrl, json, label } = options;
    const { renderCommand: renderCommand2, runCommand: runCommand2 } = await import("./render-command-3NAFANF3.js");
    if (json) {
      return runCommand2(command, services, cliVersion, async (model) => {
        const { nonInteractivePreview } = await import("./preview-FBHZI6A4.js");
        return nonInteractivePreview(services, model, { baseUrl, label });
      });
    }
    const { PreviewCommand } = await import("./PreviewCommand-A72N3DEE.js");
    await renderCommand2(
      command,
      services,
      cliVersion,
      /* @__PURE__ */ import_react4.default.createElement(PreviewCommand, { baseUrl, label, environment: "preview" })
    );
  });
};

// ../cli-astro-commands/src/commands/promote.tsx
init_esm_shims();
var import_react5 = __toESM(require_react(), 1);
var promoteCommand = (services, cliVersion) => {
  return createCommand("promote").addOption(
    new Option(
      "--base-url <url>",
      "The base url for when static files have been uploaded to an external cdn"
    ).argParser(validateBase).hideHelp()
  ).addOption(
    new Option("-l, --label <label>", "A custom label for this preview").argParser(validateLabel).hideHelp()
  ).action(async (options, command) => {
    const { baseUrl, label } = options;
    const { renderCommand: renderCommand2 } = await import("./render-command-3NAFANF3.js");
    const { PromoteCommand } = await import("./PromoteCommand-QAOZIIXY.js");
    await renderCommand2(
      command,
      services,
      cliVersion,
      /* @__PURE__ */ import_react5.default.createElement(PromoteCommand, { baseUrl, label })
    );
  });
};

// ../cli-astro-commands/src/commands/release.tsx
init_esm_shims();
var import_react6 = __toESM(require_react(), 1);
var releaseCommand = (services, cliVersion) => {
  return createCommand("release").description("Release a new version of your application").addOption(
    new Option(
      "--base-url <url>",
      "The base url for when static files have been uploaded to an external cdn"
    ).argParser(validateBase).hideHelp()
  ).addOption(
    new Option("-l, --label <label>", "A custom label for this preview").argParser(validateLabel).hideHelp()
  ).option(
    "-c, --comment <comment>",
    "The comment for your release version, not visible by users",
    validateReleaseComment
  ).addOption(
    new Option(
      "-t, --version-type <type>",
      "The type of version you would like to create"
    ).choices(["major", "minor"])
  ).action(async (options, command) => {
    const { renderCommand: renderCommand2 } = await import("./render-command-3NAFANF3.js");
    const { ReleaseCommand } = await import("./ReleaseCommand-JHFUWMKH.js");
    const { baseUrl, label, comment } = options;
    const versionType = options.versionType ? {
      major: ReleaseVersionType.MAJOR,
      minor: ReleaseVersionType.MINOR
    }[options.versionType] : void 0;
    await renderCommand2(
      command,
      services,
      cliVersion,
      /* @__PURE__ */ import_react6.default.createElement(
        ReleaseCommand,
        {
          baseUrl,
          label,
          comment,
          versionType
        }
      )
    );
  });
};

// ../cli-astro-commands/src/commands/dev.tsx
init_esm_shims();
var import_react7 = __toESM(require_react(), 1);
var devCommand = (services, cliVersion) => {
  return createCommand("dev").option(
    "--port <port>",
    "Specify which port to run on. Defaults to 4321.",
    (value) => {
      return parseInt(value, 10);
    }
  ).option(
    "--allowed-hosts <allowedHosts>",
    "Specify a comma-separated list of allowed hosts or allow any hostname."
  ).addOption(
    new Option(
      "--base-url <url>",
      "The base url for when static files have been uploaded to an external cdn"
    ).argParser(validateBase).hideHelp()
  ).action(async (options, command) => {
    const { DevCommand } = await import("./DevCommand-N5CAC2JX.js");
    const { renderCommand: renderCommand2 } = await import("./render-command-3NAFANF3.js");
    await renderCommand2(
      command,
      services,
      cliVersion,
      /* @__PURE__ */ import_react7.default.createElement(
        DevCommand,
        {
          port: options.port,
          allowedHosts: options.allowedHosts,
          serverExposedUrl: options.baseUrl
        }
      )
    );
  });
};

// ../cli-astro-commands/src/commands/generate.tsx
init_esm_shims();
var import_react8 = __toESM(require_react(), 1);
var generateCommand = (services, cliVersion) => {
  return createCommand("generate").description("Generate a new extension for your project").option(
    "--type <type>",
    "Extension type to generate",
    validateDocsExtensionType
  ).action(async (options, command) => {
    const { GenerateCommand } = await import("./GenerateCommand-ZP7EABEB.js");
    const { renderCommand: renderCommand2 } = await import("./render-command-3NAFANF3.js");
    const { type } = options;
    await renderCommand2(
      command,
      services,
      cliVersion,
      /* @__PURE__ */ import_react8.default.createElement(GenerateCommand, { docsExtensionType: type })
    );
  });
};

// ../cli-astro-commands/src/commands/translation.tsx
init_esm_shims();

// ../cli-astro-commands/src/commands/translation-pull.tsx
init_esm_shims();
var import_react9 = __toESM(require_react(), 1);
var translationPullCommand = (services, cliVersion) => {
  return createCommand("pull").description(
    "Pulls site translation files and saves them to your project directory"
  ).addOption(
    new Option(NO_TTY_JSON_OUTPUT_OPTION, "Output in JSON format").hideHelp()
  ).action(async (opts, command) => {
    if (opts.json) {
      return runCommand(command, services, cliVersion, async (model) => {
        const { nonInteractiveTranslationPull } = await import("./translation-pull-DYZH7TEM.js");
        return nonInteractiveTranslationPull(services, model);
      });
    }
    const { TranslationPullCommand } = await import("./TranslationPullCommand-7UBXP3K2.js");
    await renderCommand(
      command,
      services,
      cliVersion,
      /* @__PURE__ */ import_react9.default.createElement(TranslationPullCommand, null)
    );
  });
};

// ../cli-astro-commands/src/commands/translation.tsx
var translationCommand = (services, cliVersion) => {
  return createCommand("translation").description("Work with site translations").addCommand(translationPullCommand(services, cliVersion));
};

// ../cli-astro-commands/src/commands/root-command.ts
var getRootCommand = async (services, cliVersion) => {
  return createCommand("astro").enablePositionalOptions(true).addCommand(buildCommand(services, cliVersion)).addCommand(envCommand(services, cliVersion)).addCommand(previewCommand(services, cliVersion)).addCommand(promoteCommand(services, cliVersion), { hidden: true }).addCommand(releaseCommand(services, cliVersion)).addCommand(devCommand(services, cliVersion)).addCommand(generateCommand(services, cliVersion)).addCommand(translationCommand(services, cliVersion), { hidden: true });
};
export {
  getRootCommand
};
