// @wix/cli/build/uninstall-LK4CBF6D.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  resolveNpmDependencies,
  resolvePackageManager
} from "./chunk-6UNOO4VU.js";
import {
  createLogger
} from "./chunk-5V3Y5VKA.js";
import "./chunk-5AY2TF4R.js";
import "./chunk-DXSZXJSY.js";
import {
  createI18nT
} from "./chunk-SBOFW7ZK.js";
import "./chunk-HEANTKSX.js";
import "./chunk-CH52GF36.js";
import "./chunk-6EJNAOBP.js";
import {
  authenticate
} from "./chunk-MU5ZFQBS.js";
import "./chunk-B5TSS2PZ.js";
import "./chunk-IRKCE3VI.js";
import "./chunk-AT5WRPKM.js";
import "./chunk-2R5BII2J.js";
import "./chunk-W54PG6O5.js";
import "./chunk-YLWTNITT.js";
import "./chunk-ZWLXMEDA.js";
import "./chunk-3Q7XZST5.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-6WEX76B7.js";
import "./chunk-KJ37XZQA.js";
import "./chunk-3SLQR6EP.js";
import "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-TKJGN2XT.js";
import "./chunk-C5PTSUGF.js";
import "./chunk-WDXANXXO.js";
import "./chunk-2OYCKIQP.js";
import "./chunk-F43XHKVL.js";
import {
  CliError,
  CliErrorCode
} from "./chunk-HNXJYLVA.js";
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site-old/src/uninstall/index.ts
init_esm_shims();

// ../cli-site-old/src/uninstall/uninstall.ts
init_esm_shims();
async function uninstall(model, services, options, initialPackageName) {
  const t = createI18nT();
  const logger = createLogger({ t });
  const authState = await authenticate({
    metaSiteId: model.metaSiteId,
    biLogger: services.biLogger,
    errorReporter: services.errorReporter
  });
  const packageName = initialPackageName.trim();
  if (!packageName) {
    throw new CliError({
      code: CliErrorCode.WixUninstallMissingPackageName(),
      cause: null
    });
  }
  const logDependencyUninstalled = logger.logDependencyUninstalling(packageName);
  try {
    const packageManager = await resolvePackageManager(
      model.projectFolder,
      options
    );
    await packageManager.runUninstallPackage(model.projectFolder, packageName);
    await resolveNpmDependencies(model.projectFolder, { authState });
    logDependencyUninstalled.success();
  } catch (error) {
    logDependencyUninstalled.fail();
    throw error;
  }
}
export {
  uninstall
};
